{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: thefuzz in /usr/local/python/3.10.8/lib/python3.10/site-packages (0.20.0)\n",
      "Requirement already satisfied: rapidfuzz<4.0.0,>=3.0.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from thefuzz) (3.5.2)\n",
      "Requirement already satisfied: nltk in /usr/local/python/3.10.8/lib/python3.10/site-packages (3.8.1)\n",
      "Requirement already satisfied: click in /usr/local/python/3.10.8/lib/python3.10/site-packages (from nltk) (8.1.7)\n",
      "Requirement already satisfied: joblib in /home/codespace/.local/lib/python3.10/site-packages (from nltk) (1.3.2)\n",
      "Requirement already satisfied: regex>=2021.8.3 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from nltk) (2023.10.3)\n",
      "Requirement already satisfied: tqdm in /usr/local/python/3.10.8/lib/python3.10/site-packages (from nltk) (4.64.0)\n",
      "Requirement already satisfied: spacy in /usr/local/python/3.10.8/lib/python3.10/site-packages (3.6.1)\n",
      "Requirement already satisfied: spacy-legacy<3.1.0,>=3.0.11 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (3.0.12)\n",
      "Requirement already satisfied: spacy-loggers<2.0.0,>=1.0.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (1.0.5)\n",
      "Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (1.0.10)\n",
      "Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (2.0.8)\n",
      "Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (3.0.9)\n",
      "Requirement already satisfied: thinc<8.2.0,>=8.1.8 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (8.1.12)\n",
      "Requirement already satisfied: wasabi<1.2.0,>=0.9.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (1.1.2)\n",
      "Requirement already satisfied: srsly<3.0.0,>=2.4.3 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (2.4.8)\n",
      "Requirement already satisfied: catalogue<2.1.0,>=2.0.6 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (2.0.10)\n",
      "Requirement already satisfied: typer<0.10.0,>=0.3.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (0.9.0)\n",
      "Requirement already satisfied: pathy>=0.10.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (0.10.3)\n",
      "Requirement already satisfied: smart-open<7.0.0,>=5.2.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (6.4.0)\n",
      "Requirement already satisfied: tqdm<5.0.0,>=4.38.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (4.64.0)\n",
      "Requirement already satisfied: numpy>=1.15.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (1.24.2)\n",
      "Requirement already satisfied: requests<3.0.0,>=2.13.0 in /home/codespace/.local/lib/python3.10/site-packages (from spacy) (2.31.0)\n",
      "Requirement already satisfied: pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (2.4.2)\n",
      "Requirement already satisfied: jinja2 in /home/codespace/.local/lib/python3.10/site-packages (from spacy) (3.1.2)\n",
      "Requirement already satisfied: setuptools in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (68.0.0)\n",
      "Requirement already satisfied: packaging>=20.0 in /home/codespace/.local/lib/python3.10/site-packages (from spacy) (23.2)\n",
      "Requirement already satisfied: langcodes<4.0.0,>=3.2.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy) (3.3.0)\n",
      "Requirement already satisfied: annotated-types>=0.4.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4->spacy) (0.6.0)\n",
      "Requirement already satisfied: pydantic-core==2.10.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4->spacy) (2.10.1)\n",
      "Requirement already satisfied: typing-extensions>=4.6.1 in /home/codespace/.local/lib/python3.10/site-packages (from pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4->spacy) (4.8.0)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /home/codespace/.local/lib/python3.10/site-packages (from requests<3.0.0,>=2.13.0->spacy) (3.3.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /home/codespace/.local/lib/python3.10/site-packages (from requests<3.0.0,>=2.13.0->spacy) (3.4)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /home/codespace/.local/lib/python3.10/site-packages (from requests<3.0.0,>=2.13.0->spacy) (2.0.7)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /home/codespace/.local/lib/python3.10/site-packages (from requests<3.0.0,>=2.13.0->spacy) (2023.7.22)\n",
      "Requirement already satisfied: blis<0.8.0,>=0.7.8 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from thinc<8.2.0,>=8.1.8->spacy) (0.7.11)\n",
      "Requirement already satisfied: confection<1.0.0,>=0.0.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from thinc<8.2.0,>=8.1.8->spacy) (0.1.3)\n",
      "Requirement already satisfied: click<9.0.0,>=7.1.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from typer<0.10.0,>=0.3.0->spacy) (8.1.7)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /home/codespace/.local/lib/python3.10/site-packages (from jinja2->spacy) (2.1.3)\n",
      "Requirement already satisfied: scispacy in /usr/local/python/3.10.8/lib/python3.10/site-packages (0.5.3)\n",
      "Requirement already satisfied: spacy<3.7.0,>=3.6.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from scispacy) (3.6.1)\n",
      "Requirement already satisfied: scipy<1.11 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from scispacy) (1.10.1)\n",
      "Requirement already satisfied: requests<3.0.0,>=2.0.0 in /home/codespace/.local/lib/python3.10/site-packages (from scispacy) (2.31.0)\n",
      "Requirement already satisfied: conllu in /usr/local/python/3.10.8/lib/python3.10/site-packages (from scispacy) (4.5.3)\n",
      "Requirement already satisfied: numpy in /usr/local/python/3.10.8/lib/python3.10/site-packages (from scispacy) (1.24.2)\n",
      "Requirement already satisfied: joblib in /home/codespace/.local/lib/python3.10/site-packages (from scispacy) (1.3.2)\n",
      "Requirement already satisfied: nmslib>=1.7.3.6 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from scispacy) (2.1.1)\n",
      "Requirement already satisfied: scikit-learn>=0.20.3 in /home/codespace/.local/lib/python3.10/site-packages (from scispacy) (1.3.2)\n",
      "Requirement already satisfied: pysbd in /usr/local/python/3.10.8/lib/python3.10/site-packages (from scispacy) (0.3.4)\n",
      "Requirement already satisfied: pybind11<2.6.2 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from nmslib>=1.7.3.6->scispacy) (2.6.1)\n",
      "Requirement already satisfied: psutil in /home/codespace/.local/lib/python3.10/site-packages (from nmslib>=1.7.3.6->scispacy) (5.9.6)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /home/codespace/.local/lib/python3.10/site-packages (from requests<3.0.0,>=2.0.0->scispacy) (3.3.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /home/codespace/.local/lib/python3.10/site-packages (from requests<3.0.0,>=2.0.0->scispacy) (3.4)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /home/codespace/.local/lib/python3.10/site-packages (from requests<3.0.0,>=2.0.0->scispacy) (2.0.7)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /home/codespace/.local/lib/python3.10/site-packages (from requests<3.0.0,>=2.0.0->scispacy) (2023.7.22)\n",
      "Requirement already satisfied: threadpoolctl>=2.0.0 in /home/codespace/.local/lib/python3.10/site-packages (from scikit-learn>=0.20.3->scispacy) (3.2.0)\n",
      "Requirement already satisfied: spacy-legacy<3.1.0,>=3.0.11 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (3.0.12)\n",
      "Requirement already satisfied: spacy-loggers<2.0.0,>=1.0.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (1.0.5)\n",
      "Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (1.0.10)\n",
      "Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (2.0.8)\n",
      "Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (3.0.9)\n",
      "Requirement already satisfied: thinc<8.2.0,>=8.1.8 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (8.1.12)\n",
      "Requirement already satisfied: wasabi<1.2.0,>=0.9.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (1.1.2)\n",
      "Requirement already satisfied: srsly<3.0.0,>=2.4.3 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (2.4.8)\n",
      "Requirement already satisfied: catalogue<2.1.0,>=2.0.6 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (2.0.10)\n",
      "Requirement already satisfied: typer<0.10.0,>=0.3.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (0.9.0)\n",
      "Requirement already satisfied: pathy>=0.10.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (0.10.3)\n",
      "Requirement already satisfied: smart-open<7.0.0,>=5.2.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (6.4.0)\n",
      "Requirement already satisfied: tqdm<5.0.0,>=4.38.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (4.64.0)\n",
      "Requirement already satisfied: pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (2.4.2)\n",
      "Requirement already satisfied: jinja2 in /home/codespace/.local/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (3.1.2)\n",
      "Requirement already satisfied: setuptools in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (68.0.0)\n",
      "Requirement already satisfied: packaging>=20.0 in /home/codespace/.local/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (23.2)\n",
      "Requirement already satisfied: langcodes<4.0.0,>=3.2.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.0->scispacy) (3.3.0)\n",
      "Requirement already satisfied: annotated-types>=0.4.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4->spacy<3.7.0,>=3.6.0->scispacy) (0.6.0)\n",
      "Requirement already satisfied: pydantic-core==2.10.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4->spacy<3.7.0,>=3.6.0->scispacy) (2.10.1)\n",
      "Requirement already satisfied: typing-extensions>=4.6.1 in /home/codespace/.local/lib/python3.10/site-packages (from pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4->spacy<3.7.0,>=3.6.0->scispacy) (4.8.0)\n",
      "Requirement already satisfied: blis<0.8.0,>=0.7.8 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from thinc<8.2.0,>=8.1.8->spacy<3.7.0,>=3.6.0->scispacy) (0.7.11)\n",
      "Requirement already satisfied: confection<1.0.0,>=0.0.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from thinc<8.2.0,>=8.1.8->spacy<3.7.0,>=3.6.0->scispacy) (0.1.3)\n",
      "Requirement already satisfied: click<9.0.0,>=7.1.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from typer<0.10.0,>=0.3.0->spacy<3.7.0,>=3.6.0->scispacy) (8.1.7)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /home/codespace/.local/lib/python3.10/site-packages (from jinja2->spacy<3.7.0,>=3.6.0->scispacy) (2.1.3)\n",
      "Collecting https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.5.3/en_core_sci_lg-0.5.3.tar.gz\n",
      "  Using cached https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.5.3/en_core_sci_lg-0.5.3.tar.gz (531.2 MB)\n",
      "  Installing build dependencies ... \u001b[?25ldone\n",
      "\u001b[?25h  Getting requirements to build wheel ... \u001b[?25ldone\n",
      "\u001b[?25h  Preparing metadata (pyproject.toml) ... \u001b[?25ldone\n",
      "\u001b[?25hRequirement already satisfied: spacy<3.7.0,>=3.6.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from en-core-sci-lg==0.5.3) (3.6.1)\n",
      "Requirement already satisfied: spacy-legacy<3.1.0,>=3.0.11 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (3.0.12)\n",
      "Requirement already satisfied: spacy-loggers<2.0.0,>=1.0.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (1.0.5)\n",
      "Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (1.0.10)\n",
      "Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (2.0.8)\n",
      "Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (3.0.9)\n",
      "Requirement already satisfied: thinc<8.2.0,>=8.1.8 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (8.1.12)\n",
      "Requirement already satisfied: wasabi<1.2.0,>=0.9.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (1.1.2)\n",
      "Requirement already satisfied: srsly<3.0.0,>=2.4.3 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (2.4.8)\n",
      "Requirement already satisfied: catalogue<2.1.0,>=2.0.6 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (2.0.10)\n",
      "Requirement already satisfied: typer<0.10.0,>=0.3.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (0.9.0)\n",
      "Requirement already satisfied: pathy>=0.10.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (0.10.3)\n",
      "Requirement already satisfied: smart-open<7.0.0,>=5.2.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (6.4.0)\n",
      "Requirement already satisfied: tqdm<5.0.0,>=4.38.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (4.64.0)\n",
      "Requirement already satisfied: numpy>=1.15.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (1.24.2)\n",
      "Requirement already satisfied: requests<3.0.0,>=2.13.0 in /home/codespace/.local/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (2.31.0)\n",
      "Requirement already satisfied: pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (2.4.2)\n",
      "Requirement already satisfied: jinja2 in /home/codespace/.local/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (3.1.2)\n",
      "Requirement already satisfied: setuptools in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (68.0.0)\n",
      "Requirement already satisfied: packaging>=20.0 in /home/codespace/.local/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (23.2)\n",
      "Requirement already satisfied: langcodes<4.0.0,>=3.2.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (3.3.0)\n",
      "Requirement already satisfied: annotated-types>=0.4.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4->spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (0.6.0)\n",
      "Requirement already satisfied: pydantic-core==2.10.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4->spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (2.10.1)\n",
      "Requirement already satisfied: typing-extensions>=4.6.1 in /home/codespace/.local/lib/python3.10/site-packages (from pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4->spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (4.8.0)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /home/codespace/.local/lib/python3.10/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (3.3.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /home/codespace/.local/lib/python3.10/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (3.4)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /home/codespace/.local/lib/python3.10/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (2.0.7)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /home/codespace/.local/lib/python3.10/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (2023.7.22)\n",
      "Requirement already satisfied: blis<0.8.0,>=0.7.8 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from thinc<8.2.0,>=8.1.8->spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (0.7.11)\n",
      "Requirement already satisfied: confection<1.0.0,>=0.0.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from thinc<8.2.0,>=8.1.8->spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (0.1.3)\n",
      "Requirement already satisfied: click<9.0.0,>=7.1.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from typer<0.10.0,>=0.3.0->spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (8.1.7)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /home/codespace/.local/lib/python3.10/site-packages (from jinja2->spacy<3.7.0,>=3.6.1->en-core-sci-lg==0.5.3) (2.1.3)\n",
      "Requirement already satisfied: openai in /usr/local/python/3.10.8/lib/python3.10/site-packages (1.2.3)\n",
      "Requirement already satisfied: anyio<4,>=3.5.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from openai) (3.7.1)\n",
      "Requirement already satisfied: distro<2,>=1.7.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from openai) (1.8.0)\n",
      "Requirement already satisfied: httpx<1,>=0.23.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from openai) (0.25.1)\n",
      "Requirement already satisfied: pydantic<3,>=1.9.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from openai) (2.4.2)\n",
      "Requirement already satisfied: tqdm>4 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from openai) (4.64.0)\n",
      "Requirement already satisfied: typing-extensions<5,>=4.5 in /home/codespace/.local/lib/python3.10/site-packages (from openai) (4.8.0)\n",
      "Requirement already satisfied: idna>=2.8 in /home/codespace/.local/lib/python3.10/site-packages (from anyio<4,>=3.5.0->openai) (3.4)\n",
      "Requirement already satisfied: sniffio>=1.1 in /home/codespace/.local/lib/python3.10/site-packages (from anyio<4,>=3.5.0->openai) (1.3.0)\n",
      "Requirement already satisfied: exceptiongroup in /home/codespace/.local/lib/python3.10/site-packages (from anyio<4,>=3.5.0->openai) (1.1.3)\n",
      "Requirement already satisfied: certifi in /home/codespace/.local/lib/python3.10/site-packages (from httpx<1,>=0.23.0->openai) (2023.7.22)\n",
      "Requirement already satisfied: httpcore in /usr/local/python/3.10.8/lib/python3.10/site-packages (from httpx<1,>=0.23.0->openai) (1.0.2)\n",
      "Requirement already satisfied: annotated-types>=0.4.0 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from pydantic<3,>=1.9.0->openai) (0.6.0)\n",
      "Requirement already satisfied: pydantic-core==2.10.1 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from pydantic<3,>=1.9.0->openai) (2.10.1)\n",
      "Requirement already satisfied: h11<0.15,>=0.13 in /usr/local/python/3.10.8/lib/python3.10/site-packages (from httpcore->httpx<1,>=0.23.0->openai) (0.14.0)\n"
     ]
    }
   ],
   "source": [
    "!pip install thefuzz\n",
    "!pip install nltk\n",
    "!pip install spacy\n",
    "!pip install scispacy\n",
    "!pip install https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.5.3/en_core_sci_lg-0.5.3.tar.gz\n",
    "!pip install openai\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package punkt to /home/codespace/nltk_data...\n",
      "[nltk_data]   Package punkt is already up-to-date!\n",
      "[nltk_data] Downloading package wordnet to\n",
      "[nltk_data]     /home/codespace/nltk_data...\n",
      "[nltk_data]   Package wordnet is already up-to-date!\n"
     ]
    }
   ],
   "source": [
    "import nltk\n",
    "from load_data import load_ann, load_txt\n",
    "import pandas as pd\n",
    "from thefuzz import fuzz\n",
    "from nltk.stem import WordNetLemmatizer\n",
    "import spacy\n",
    "from scispacy.abbreviation import AbbreviationDetector\n",
    "from openai import OpenAI\n",
    "from scipy.spatial.distance import cosine\n",
    "from scispacy.linking import EntityLinker\n",
    "# from Bio_Epidemiology_NER.bio_recognizer import ner_prediction\n",
    "\n",
    "\n",
    "nltk.download('punkt')\n",
    "nltk.download('wordnet')\n",
    "\n",
    "lemmatizer = WordNetLemmatizer()\n",
    "\n",
    "# abreviation_handler = spacy.load(\"en_core_sci_lg\")\n",
    "# abreviation_handler.add_pipe(\"abbreviation_detector\")\n",
    "\n",
    "\n",
    "# abreviation_handler = spacy.load(\"en_core_sci_lg\")\n",
    "# abreviation_handler.add_pipe(\"abbreviation_detector\")\n",
    "\n",
    "# TODO: Put in config\n",
    "client = OpenAI(api_key='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')\n",
    "\n",
    "def get_embedding(text, model=\"text-embedding-ada-002\"):\n",
    "   text = text.replace(\"\\n\", \" \")\n",
    "   returnVal = client.embeddings.create(input = [text], model=model).data[0].embedding\n",
    "   print(returnVal)\n",
    "   return returnVal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Full data path: /workspaces/codespaces-jupyter/Project/RawData/\n",
      "Time taken to read .txt files: 0.9435818195343018\n",
      "Time taken to read .ann files and extract all metadata: 0.1653289794921875\n"
     ]
    }
   ],
   "source": [
    "PATH_TO_ZIP = \"/workspaces/codespaces-jupyter/Project/RawData\"\n",
    "DATA_PATH = f\"{PATH_TO_ZIP}/\"\n",
    "print(f\"Full data path: {DATA_PATH}\")\n",
    "# read in txt files\n",
    "txt_df = load_txt(DATA_PATH)\n",
    "# read in REASONS entities from .ann files\n",
    "ent_df, rel_df = load_ann(DATA_PATH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"\\n4. Add definition of abbreviations to 'text' in txt_df\\n\""
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"\n",
    "DATA CLEANING\n",
    "\"\"\"\n",
    "\n",
    "\"\"\"\n",
    "1. Convert 'text' columns to lowercase, in order to facilitate comparison.\n",
    "\"\"\"\n",
    "# To lowercase 'text' column in ent_df\n",
    "ent_df['text'] = ent_df['text'].str.lower()\n",
    "ent_df['text'] = ent_df['text'].str.strip()\n",
    "\n",
    "# To lowercase 'text' column in txt_df\n",
    "# txt_df['text'] = txt_df['text'].str.lower()\n",
    "\n",
    "\"\"\"\n",
    "2. Remove \\n ending from 'text' column in ent_df and in 'entity2' column in rel_df.\n",
    "\"\"\"\n",
    "ent_df['text'] = ent_df['text'].str.rstrip('\\n')\n",
    "rel_df['entity2'] = rel_df['entity2'].str.rstrip('\\n')\n",
    "\n",
    "\"\"\"\n",
    "3. Convert 'start_idx' and 'end_idx' columns in ent_df to int.\n",
    "\"\"\"\n",
    "# Drop rows that cannot be converted to int TODO: Make this better\n",
    "ent_df = ent_df[ent_df['start_idx'].str.isnumeric()]\n",
    "ent_df = ent_df[ent_df['end_idx'].str.isnumeric()]\n",
    "ent_df['start_idx'] = ent_df['start_idx'].astype(int)\n",
    "ent_df['end_idx'] = ent_df['end_idx'].astype(int)\n",
    "\n",
    "# Make new column with lemmatized text of 'text' column called 'lemmatized_text'\n",
    "ent_df['orig_txt'] = ent_df['text']\n",
    "def lemmatize_text(text):\n",
    "    words = nltk.word_tokenize(text)\n",
    "    lemmatized_words = [lemmatizer.lemmatize(word) for word in words]\n",
    "    return ' '.join(lemmatized_words)\n",
    "\n",
    "ent_df['text'] = ent_df['text'].apply(lemmatize_text)\n",
    "# Convert ent_df['text'] to string\n",
    "ent_df['text'] = ent_df['text'].astype(str)\n",
    "\n",
    "\n",
    "\"\"\"\n",
    "4. Add definition of abbreviations to 'text' in txt_df\n",
    "\"\"\"\n",
    "# Loop through each row in txt_df and then use abreviation_handler to find abbreviations and then insert definition in parenthesis after abbreviation, for every abbreviation found in the 'text' column.\n",
    "# for index, row in txt_df.iterrows():\n",
    "#     doc = abreviation_handler(row['text'])\n",
    "#     print(len(doc._.abbreviations))\n",
    "    # for abrv in doc._.abbreviations:\n",
    "    #     row['text'] = row['text'].replace(f\"{abrv}\", f\"{abrv} ({abrv._.long_form})\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "75mg\n",
      "75mg\n",
      "75mg\n",
      "106575\n",
      "106575\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "134445\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "111542\n",
      "149687\n",
      "149687\n",
      "149687\n",
      "149687\n",
      "149687\n",
      "149687\n",
      "149687\n",
      "149687\n",
      "149687\n",
      "108949\n",
      "176106\n",
      "176106\n",
      "176106\n",
      "176106\n",
      "176106\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "125867\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "174143\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n",
      "102053\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "FEATURE ENGINEERING\n",
    "\"\"\"\n",
    "\n",
    "\"\"\"\n",
    "1. Join the appropriate entity1 and entity2 for each relation in rel_df.\n",
    "\"\"\"\n",
    "# Remove first 5 letters from 'entity1' and 'entity2' column in rel_df\n",
    "rel_df['entity1'] = rel_df['entity1'].str[5:]\n",
    "rel_df['entity2'] = rel_df['entity2'].str[5:]\n",
    "rel_df = rel_df.merge(ent_df[['entity_id', 'text', 'file_idx']], how='left', left_on=['entity1', 'file_idx'], right_on=['entity_id', 'file_idx'])\n",
    "rel_df.rename(columns={'text': 'entity1_text'}, inplace=True)\n",
    "rel_df = rel_df.merge(ent_df[['entity_id', 'text', 'file_idx']], how='left', left_on=['entity2', 'file_idx'], right_on=['entity_id', 'file_idx'])\n",
    "rel_df.rename(columns={'text': 'entity2_text'}, inplace=True)\n",
    "rel_df.drop(columns=['entity_id_x', 'entity_id_y'], inplace=True)\n",
    "\n",
    "# Create column 'entity1_entity2' in rel_df\n",
    "rel_df['entity1_entity2'] = rel_df['entity1_text'] + rel_df['entity2_text']\n",
    "\n",
    "\"\"\"\n",
    "2. Get count of text in file_idx for each entity in ent_df. Do the same for the 'entity1_entity2' in rel_df.\n",
    "\"\"\"\n",
    "ent_df_count = ent_df.groupby(['text', 'file_idx']).size().reset_index(name='count_in_document')\n",
    "ent_df = ent_df.merge(ent_df_count, how='left', left_on=['text', 'file_idx'], right_on=['text', 'file_idx'])\n",
    "rel_df_count = rel_df.groupby(['entity1_entity2', 'file_idx']).size().reset_index(name='count_in_document')\n",
    "rel_df = rel_df.merge(rel_df_count, how='left', left_on=['entity1_entity2', 'file_idx'], right_on=['entity1_entity2', 'file_idx'])\n",
    "\n",
    "\"\"\"\n",
    "3. Create encoding to represent if entity in ent_df is in the 'Discharge Diagnosis', 'Chief Complaint', or 'History of Present Illness' section of the txt_df.\n",
    "\"\"\"\n",
    "def find_section_range(row, section_name):\n",
    "    lines = row['text'].split('\\n')\n",
    "    matches = [(i, fuzz.ratio(line.lower(), section_name.lower())) for i, line in enumerate(lines)]\n",
    "    matches.sort(key=lambda x: x[1], reverse=True)  # sort by fuzz.ratio in descending order\n",
    "    if not matches:\n",
    "        # Raise error if no match is found\n",
    "        raise ValueError(f\"Could not find section {section_name} in file {row['file_idx']}\")\n",
    "    start_line = matches[0][0]  # start of the range is the line with the highest fuzz.ratio\n",
    "    end_line = start_line\n",
    "    while end_line < len(lines) and lines[end_line].strip() != '':\n",
    "        end_line += 1\n",
    "    # calculate start and end index within the raw text\n",
    "    start_index = sum(len(line) + 1 for line in lines[:start_line])  # +1 for the newline character\n",
    "    end_index = sum(len(line) + 1 for line in lines[:end_line])  # +1 for the newline character\n",
    "    if row['file_idx'] == '100035':\n",
    "        # print(row['text'][start_index: start_index + 20], '\\n', row['text'][end_index: end_index + 20])\n",
    "        print(row['text'][10559:10563] )\n",
    "    return (start_index, end_index)\n",
    "\n",
    "txt_df['DD_Range'] = txt_df.apply(lambda row: find_section_range(row, 'Discharge Diagnosis'), axis=1)\n",
    "txt_df['CC_Range'] = txt_df.apply(lambda row: find_section_range(row, 'Chief Complaint'), axis=1)\n",
    "txt_df['HPI_Range'] = txt_df.apply(lambda row: find_section_range(row, 'History of Present Illness'), axis=1)\n",
    "\n",
    "# Join the 'DD_Range', 'CC_Range', and 'HPI_Range' columns from txt_df to ent_df\n",
    "ent_df = ent_df.merge(txt_df[['file_idx', 'DD_Range', 'CC_Range', 'HPI_Range']], how='left', left_on=['file_idx'], right_on=['file_idx'])\n",
    "\n",
    "# Loop through each entity in ent_df and check if it is in the 'Discharge Diagnosis', 'Chief Complaint', or 'History of Present Illness' section of the txt_df.\n",
    "# If it is, then add the section name to the 'section' column in ent_df.\n",
    "def find_section(row):\n",
    "    # Throw error if start_idx is greater than end_idx\n",
    "    if row['start_idx'] > row['end_idx']:\n",
    "        raise ValueError(f\"start_idx {row['start_idx']} is greater than end_idx {row['end_idx']}\")\n",
    "    # Throw error if start_idx and end_idx are in multiple sections\n",
    "    if row['start_idx'] >= row['DD_Range'][0] and row['end_idx'] <= row['DD_Range'][1] and row['start_idx'] >= row['CC_Range'][0] and row['end_idx'] <= row['CC_Range'][1]:\n",
    "        print(row['file_idx'])\n",
    "        # raise ValueError(f\"start_idx {row['start_idx']} and end_idx {row['end_idx']} are in both DD and CC\")\n",
    "    if row['start_idx'] >= row['DD_Range'][0] and row['end_idx'] <= row['DD_Range'][1] and row['start_idx'] >= row['HPI_Range'][0] and row['end_idx'] <= row['HPI_Range'][1]:\n",
    "        print(row['file_idx'])\n",
    "        # raise ValueError(f\"start_idx {row['start_idx']} and end_idx {row['end_idx']} are in both DD and HPI\")\n",
    "    if row['start_idx'] >= row['CC_Range'][0] and row['end_idx'] <= row['CC_Range'][1] and row['start_idx'] >= row['HPI_Range'][0] and row['end_idx'] <= row['HPI_Range'][1]:\n",
    "        print(row['file_idx'])\n",
    "        # raise ValueError(f\"start_idx {row['start_idx']} and end_idx {row['end_idx']} are in both CC and HPI\")\n",
    "    \n",
    "    # If start_idx and end_idx are in one section, return the section name\n",
    "    if row['start_idx'] >= row['DD_Range'][0] and row['end_idx'] <= row['DD_Range'][1]:\n",
    "        return 'Discharge Diagnosis'\n",
    "    elif row['start_idx'] >= row['CC_Range'][0] and row['end_idx'] <= row['CC_Range'][1]:\n",
    "        return 'Chief Complaint'\n",
    "    elif row['start_idx'] >= row['HPI_Range'][0] and row['end_idx'] <= row['HPI_Range'][1]:\n",
    "        return 'History of Present Illness'\n",
    "    else:\n",
    "        return 'Other'\n",
    "\n",
    "ent_df['section'] = ent_df.apply(lambda row: find_section(row), axis=1)\n",
    "# Drop DD_Range, CC_Range, and HPI_Range columns from ent_df\n",
    "ent_df.drop(columns=['DD_Range', 'CC_Range', 'HPI_Range'], inplace=True)\n",
    "# Apply one hot encoding to 'section' column in ent_df\n",
    "ent_df = pd.get_dummies(ent_df, columns=['section'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "txt_df_subset = txt_df[txt_df['file_idx'] == '100035']\n",
    "ent_df_subset = ent_df[ent_df['file_idx'] == '100035']\n",
    "rel_df_subset = rel_df[rel_df['file_idx'] == '100035']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Get DD_Range, CC_Range, and HPI_Range for each file in txt_df_subset and store in tuples\n",
    "# DD_Range = txt_df_subset['DD_Range'].iloc[0]\n",
    "# CC_Range = txt_df_subset['CC_Range'].iloc[0]\n",
    "# HPI_Range = txt_df_subset['HPI_Range'].iloc[0]\n",
    "\n",
    "# # Create a dict that loops through each line in the txt_df_subset 'text' column in the DD_Range and stores each diagnosis as a key and the value as an embedding of the diagnosis using get_embedding from openai.\n",
    "# diagnosis_dict = {}\n",
    "# count_dict = {}\n",
    "# for line in str(txt_df_subset['text'].iloc[0][DD_Range[0]:DD_Range[1]]).split('\\n'):\n",
    "#     if len(line) > 0:\n",
    "#         diagnosis_dict[line] = get_embedding(line)\n",
    "#         count_dict[line] = 0\n",
    "\n",
    "# curr_text = str(txt_df_subset['text'].iloc[0])\n",
    "# text_to_analyze = \"\"\n",
    "# # Add the text from the CC_Range and HPI_Range to text_to_analyze\n",
    "# text_to_analyze += curr_text[CC_Range[0]:CC_Range[1]]\n",
    "# text_to_analyze += curr_text[HPI_Range[0]:HPI_Range[1]]\n",
    "\n",
    "# # For each sentence in the txt_df_subset 'text' column in the CC_Range and HPI_Range, convert the sentence to an embedding using get_embedding from openai. Get the sentences using nltk\n",
    "# for line in nltk.sent_tokenize(text_to_analyze):\n",
    "#     # For each diagnosis in diagnosis_dict, calculate the cosine similarity between the diagnosis embedding and the sentence embedding. Store the diagnosis with the highest cosine similarity in the count_dict.\n",
    "#     max_sim = 0\n",
    "#     max_diagnosis = \"\"\n",
    "#     for diagnosis, embedding in diagnosis_dict.items():\n",
    "#         similarity = cosine(embedding, get_embedding(line))\n",
    "#         if similarity > max_sim:\n",
    "#             max_diagnosis = diagnosis\n",
    "#             max_sim = similarity\n",
    "#     print(max_sim)\n",
    "#     count_dict[max_diagnosis] += 1\n",
    "\n",
    "# print(count_dict)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# nlp = spacy.load(\"en_core_sci_lg\")\n",
    "# nlp.add_pipe(\"scispacy_linker\", config={\"resolve_abbreviations\": True, \"linker_name\": \"umls\"})\n",
    "\n",
    "# diagnosis_dict = {}\n",
    "# objects = []\n",
    "# for line in str(txt_df_subset['text'].iloc[0][DD_Range[0]:DD_Range[1]]).split('\\n'):\n",
    "#     if len(line) > 0:\n",
    "#         doc = nlp(line)\n",
    "#         linker = nlp.get_pipe(\"scispacy_linker\")\n",
    "#         for entity in doc.ents:\n",
    "#             for umls_ent in entity._.kb_ents:\n",
    "#                 try:\n",
    "#                     objects.append(linker.kb.cui_to_entity[umls_ent[0]])\n",
    "#                     break\n",
    "#                 except:\n",
    "#                     print(\"error\")\n",
    "#                 # diagnosis_dict[ent.entity_id] = ent.canonical_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for object in objects:\n",
    "#     print(object[2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/codespace/.python/current/lib/python3.10/site-packages/spacy/language.py:2141: FutureWarning: Possible set union at position 6328\n",
      "  deserializers[\"tokenizer\"] = lambda p: self.tokenizer.from_disk(  # type: ignore[union-attr]\n",
      "/home/codespace/.local/lib/python3.10/site-packages/sklearn/base.py:348: InconsistentVersionWarning: Trying to unpickle estimator TfidfTransformer from version 1.1.2 when using version 1.3.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/codespace/.local/lib/python3.10/site-packages/sklearn/base.py:348: InconsistentVersionWarning: Trying to unpickle estimator TfidfVectorizer from version 1.1.2 when using version 1.3.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "ename": "",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31mThe Kernel crashed while executing code in the the current cell or a previous cell. Please review the code in the cell(s) to identify a possible cause of the failure. Click <a href='https://aka.ms/vscodeJupyterKernelCrash'>here</a> for more info. View Jupyter <a href='command:jupyter.viewOutput'>log</a> for further details."
     ]
    }
   ],
   "source": [
    "df = pd.DataFrame(columns=['file_idx', 'primary_diagnosis', 'count'])  # initialize df as an empty DataFrame\n",
    "def create_freq_dict(df_input):\n",
    "    nlp = spacy.load(\"en_core_sci_lg\")\n",
    "    nlp.add_pipe(\"scispacy_linker\", config={\"resolve_abbreviations\": True, \"linker_name\": \"umls\"})\n",
    "    linker = nlp.get_pipe(\"scispacy_linker\")\n",
    "\n",
    "    for i, file_idx in enumerate(txt_df['file_idx'].unique()):\n",
    "        print(i, \" out of \", len(txt_df['file_idx'].unique()))\n",
    "        txt_df_subset = txt_df[txt_df['file_idx'] == file_idx]\n",
    "\n",
    "        # Get DD_Range, CC_Range, and HPI_Range for each file in txt_df_subset and store in tuples\n",
    "        DD_Range = txt_df_subset['DD_Range'].iloc[0]\n",
    "        CC_Range = txt_df_subset['CC_Range'].iloc[0]\n",
    "        HPI_Range = txt_df_subset['HPI_Range'].iloc[0]\n",
    "\n",
    "        # Create a dict that loops through each line in the txt_df_subset 'text' column in the DD_Range and stores each diagnosis as a key and the value as an embedding of the diagnosis using get_embedding from openai.\n",
    "        diagnosis_dict = {}\n",
    "        for i, line in enumerate(str(txt_df_subset['text'].iloc[0][DD_Range[0]:DD_Range[1]]).split('\\n')):\n",
    "            if i == 0:\n",
    "                continue\n",
    "            if len(line) > 0:\n",
    "                doc = nlp(line)\n",
    "                for entity in doc.ents:\n",
    "                    diagnosis_dict[entity] = []\n",
    "                    for umls_ent in entity._.kb_ents:\n",
    "                        try:\n",
    "                            object = linker.kb.cui_to_entity[umls_ent[0]]\n",
    "                            for alias in object[2]:\n",
    "                                diagnosis_dict[entity].append(alias)\n",
    "                        except Exception as e:\n",
    "                            print(e)\n",
    "\n",
    "        count_dict = {}\n",
    "        for key, _ in diagnosis_dict.items():\n",
    "            count_dict[key] = 0\n",
    "\n",
    "        curr_text = str(txt_df_subset['text'].iloc[0])\n",
    "        # text_to_analyze = \"\"\n",
    "        # # Add the text from the CC_Range and HPI_Range to text_to_analyze\n",
    "        # text_to_analyze += curr_text[CC_Range[0]:CC_Range[1]]\n",
    "        # text_to_analyze += curr_text[HPI_Range[0]:HPI_Range[1]]\n",
    "\n",
    "        for line in nltk.sent_tokenize(curr_text):\n",
    "            doc = nlp(line)\n",
    "            for entity in doc.ents:\n",
    "                for umls_ent in entity._.kb_ents:\n",
    "                    try:\n",
    "                        object = linker.kb.cui_to_entity[umls_ent[0]]\n",
    "                        for key, arr in diagnosis_dict.items():\n",
    "                            for alias in object[2]:\n",
    "                                if alias in arr:\n",
    "                                    count_dict[key] += 1\n",
    "                                    break\n",
    "                    except Exception as e:\n",
    "                        print(e)\n",
    "\n",
    "        for key, value in count_dict.items():\n",
    "            df_input = df_input.append({'file_idx': file_idx, 'primary_diagnosis': key, 'count': value}, ignore_index=True)\n",
    "\n",
    "    return df_input\n",
    "\n",
    "df = create_freq_dict(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save df to csv\n",
    "df.to_csv('primary_diagnosis.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m/workspaces/codespaces-jupyter/Project/Dataset Creation/dataset_creation.ipynb Cell 11\u001b[0m line \u001b[0;36m7\n\u001b[1;32m      <a href='vscode-notebook-cell://codespaces%2Bspecial-space-orbit-q9x6v659q5w3r9v/workspaces/codespaces-jupyter/Project/Dataset%20Creation/dataset_creation.ipynb#X12sdnNjb2RlLXJlbW90ZQ%3D%3D?line=3'>4</a>\u001b[0m HPI_Range \u001b[39m=\u001b[39m txt_df_subset[\u001b[39m'\u001b[39m\u001b[39mHPI_Range\u001b[39m\u001b[39m'\u001b[39m]\u001b[39m.\u001b[39miloc[\u001b[39m0\u001b[39m]\n\u001b[1;32m      <a href='vscode-notebook-cell://codespaces%2Bspecial-space-orbit-q9x6v659q5w3r9v/workspaces/codespaces-jupyter/Project/Dataset%20Creation/dataset_creation.ipynb#X12sdnNjb2RlLXJlbW90ZQ%3D%3D?line=5'>6</a>\u001b[0m \u001b[39m# Create a dict that loops through each line in the txt_df_subset 'text' column in the DD_Range and stores each diagnosis as a key and the value as an embedding of the diagnosis using get_embedding from openai.\u001b[39;00m\n\u001b[0;32m----> <a href='vscode-notebook-cell://codespaces%2Bspecial-space-orbit-q9x6v659q5w3r9v/workspaces/codespaces-jupyter/Project/Dataset%20Creation/dataset_creation.ipynb#X12sdnNjb2RlLXJlbW90ZQ%3D%3D?line=6'>7</a>\u001b[0m nlp \u001b[39m=\u001b[39m spacy\u001b[39m.\u001b[39;49mload(\u001b[39m\"\u001b[39;49m\u001b[39men_core_sci_lg\u001b[39;49m\u001b[39m\"\u001b[39;49m)\n\u001b[1;32m      <a href='vscode-notebook-cell://codespaces%2Bspecial-space-orbit-q9x6v659q5w3r9v/workspaces/codespaces-jupyter/Project/Dataset%20Creation/dataset_creation.ipynb#X12sdnNjb2RlLXJlbW90ZQ%3D%3D?line=7'>8</a>\u001b[0m nlp\u001b[39m.\u001b[39madd_pipe(\u001b[39m\"\u001b[39m\u001b[39mscispacy_linker\u001b[39m\u001b[39m\"\u001b[39m, config\u001b[39m=\u001b[39m{\u001b[39m\"\u001b[39m\u001b[39mresolve_abbreviations\u001b[39m\u001b[39m\"\u001b[39m: \u001b[39mTrue\u001b[39;00m, \u001b[39m\"\u001b[39m\u001b[39mlinker_name\u001b[39m\u001b[39m\"\u001b[39m: \u001b[39m\"\u001b[39m\u001b[39mumls\u001b[39m\u001b[39m\"\u001b[39m})\n\u001b[1;32m      <a href='vscode-notebook-cell://codespaces%2Bspecial-space-orbit-q9x6v659q5w3r9v/workspaces/codespaces-jupyter/Project/Dataset%20Creation/dataset_creation.ipynb#X12sdnNjb2RlLXJlbW90ZQ%3D%3D?line=8'>9</a>\u001b[0m diagnosis_dict \u001b[39m=\u001b[39m {}\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/spacy/__init__.py:51\u001b[0m, in \u001b[0;36mload\u001b[0;34m(name, vocab, disable, enable, exclude, config)\u001b[0m\n\u001b[1;32m     27\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mload\u001b[39m(\n\u001b[1;32m     28\u001b[0m     name: Union[\u001b[39mstr\u001b[39m, Path],\n\u001b[1;32m     29\u001b[0m     \u001b[39m*\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     34\u001b[0m     config: Union[Dict[\u001b[39mstr\u001b[39m, Any], Config] \u001b[39m=\u001b[39m util\u001b[39m.\u001b[39mSimpleFrozenDict(),\n\u001b[1;32m     35\u001b[0m ) \u001b[39m-\u001b[39m\u001b[39m>\u001b[39m Language:\n\u001b[1;32m     36\u001b[0m \u001b[39m    \u001b[39m\u001b[39m\"\"\"Load a spaCy model from an installed package or a local path.\u001b[39;00m\n\u001b[1;32m     37\u001b[0m \n\u001b[1;32m     38\u001b[0m \u001b[39m    name (str): Package name or model path.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     49\u001b[0m \u001b[39m    RETURNS (Language): The loaded nlp object.\u001b[39;00m\n\u001b[1;32m     50\u001b[0m \u001b[39m    \"\"\"\u001b[39;00m\n\u001b[0;32m---> 51\u001b[0m     \u001b[39mreturn\u001b[39;00m util\u001b[39m.\u001b[39;49mload_model(\n\u001b[1;32m     52\u001b[0m         name,\n\u001b[1;32m     53\u001b[0m         vocab\u001b[39m=\u001b[39;49mvocab,\n\u001b[1;32m     54\u001b[0m         disable\u001b[39m=\u001b[39;49mdisable,\n\u001b[1;32m     55\u001b[0m         enable\u001b[39m=\u001b[39;49menable,\n\u001b[1;32m     56\u001b[0m         exclude\u001b[39m=\u001b[39;49mexclude,\n\u001b[1;32m     57\u001b[0m         config\u001b[39m=\u001b[39;49mconfig,\n\u001b[1;32m     58\u001b[0m     )\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/spacy/util.py:465\u001b[0m, in \u001b[0;36mload_model\u001b[0;34m(name, vocab, disable, enable, exclude, config)\u001b[0m\n\u001b[1;32m    463\u001b[0m     \u001b[39mreturn\u001b[39;00m get_lang_class(name\u001b[39m.\u001b[39mreplace(\u001b[39m\"\u001b[39m\u001b[39mblank:\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39m\"\u001b[39m\u001b[39m\"\u001b[39m))()\n\u001b[1;32m    464\u001b[0m \u001b[39mif\u001b[39;00m is_package(name):  \u001b[39m# installed as package\u001b[39;00m\n\u001b[0;32m--> 465\u001b[0m     \u001b[39mreturn\u001b[39;00m load_model_from_package(name, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs)  \u001b[39m# type: ignore[arg-type]\u001b[39;00m\n\u001b[1;32m    466\u001b[0m \u001b[39mif\u001b[39;00m Path(name)\u001b[39m.\u001b[39mexists():  \u001b[39m# path to model data directory\u001b[39;00m\n\u001b[1;32m    467\u001b[0m     \u001b[39mreturn\u001b[39;00m load_model_from_path(Path(name), \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs)  \u001b[39m# type: ignore[arg-type]\u001b[39;00m\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/spacy/util.py:501\u001b[0m, in \u001b[0;36mload_model_from_package\u001b[0;34m(name, vocab, disable, enable, exclude, config)\u001b[0m\n\u001b[1;32m    484\u001b[0m \u001b[39m\u001b[39m\u001b[39m\"\"\"Load a model from an installed package.\u001b[39;00m\n\u001b[1;32m    485\u001b[0m \n\u001b[1;32m    486\u001b[0m \u001b[39mname (str): The package name.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    498\u001b[0m \u001b[39mRETURNS (Language): The loaded nlp object.\u001b[39;00m\n\u001b[1;32m    499\u001b[0m \u001b[39m\"\"\"\u001b[39;00m\n\u001b[1;32m    500\u001b[0m \u001b[39mcls\u001b[39m \u001b[39m=\u001b[39m importlib\u001b[39m.\u001b[39mimport_module(name)\n\u001b[0;32m--> 501\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mcls\u001b[39;49m\u001b[39m.\u001b[39;49mload(vocab\u001b[39m=\u001b[39;49mvocab, disable\u001b[39m=\u001b[39;49mdisable, enable\u001b[39m=\u001b[39;49menable, exclude\u001b[39m=\u001b[39;49mexclude, config\u001b[39m=\u001b[39;49mconfig)\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/en_core_sci_lg/__init__.py:10\u001b[0m, in \u001b[0;36mload\u001b[0;34m(**overrides)\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mload\u001b[39m(\u001b[39m*\u001b[39m\u001b[39m*\u001b[39moverrides):\n\u001b[0;32m---> 10\u001b[0m     \u001b[39mreturn\u001b[39;00m load_model_from_init_py(\u001b[39m__file__\u001b[39;49m, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49moverrides)\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/spacy/util.py:682\u001b[0m, in \u001b[0;36mload_model_from_init_py\u001b[0;34m(init_file, vocab, disable, enable, exclude, config)\u001b[0m\n\u001b[1;32m    680\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mnot\u001b[39;00m model_path\u001b[39m.\u001b[39mexists():\n\u001b[1;32m    681\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mIOError\u001b[39;00m(Errors\u001b[39m.\u001b[39mE052\u001b[39m.\u001b[39mformat(path\u001b[39m=\u001b[39mdata_path))\n\u001b[0;32m--> 682\u001b[0m \u001b[39mreturn\u001b[39;00m load_model_from_path(\n\u001b[1;32m    683\u001b[0m     data_path,\n\u001b[1;32m    684\u001b[0m     vocab\u001b[39m=\u001b[39;49mvocab,\n\u001b[1;32m    685\u001b[0m     meta\u001b[39m=\u001b[39;49mmeta,\n\u001b[1;32m    686\u001b[0m     disable\u001b[39m=\u001b[39;49mdisable,\n\u001b[1;32m    687\u001b[0m     enable\u001b[39m=\u001b[39;49menable,\n\u001b[1;32m    688\u001b[0m     exclude\u001b[39m=\u001b[39;49mexclude,\n\u001b[1;32m    689\u001b[0m     config\u001b[39m=\u001b[39;49mconfig,\n\u001b[1;32m    690\u001b[0m )\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/spacy/util.py:547\u001b[0m, in \u001b[0;36mload_model_from_path\u001b[0;34m(model_path, meta, vocab, disable, enable, exclude, config)\u001b[0m\n\u001b[1;32m    538\u001b[0m config \u001b[39m=\u001b[39m load_config(config_path, overrides\u001b[39m=\u001b[39moverrides)\n\u001b[1;32m    539\u001b[0m nlp \u001b[39m=\u001b[39m load_model_from_config(\n\u001b[1;32m    540\u001b[0m     config,\n\u001b[1;32m    541\u001b[0m     vocab\u001b[39m=\u001b[39mvocab,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    545\u001b[0m     meta\u001b[39m=\u001b[39mmeta,\n\u001b[1;32m    546\u001b[0m )\n\u001b[0;32m--> 547\u001b[0m \u001b[39mreturn\u001b[39;00m nlp\u001b[39m.\u001b[39;49mfrom_disk(model_path, exclude\u001b[39m=\u001b[39;49mexclude, overrides\u001b[39m=\u001b[39;49moverrides)\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/spacy/language.py:2155\u001b[0m, in \u001b[0;36mLanguage.from_disk\u001b[0;34m(self, path, exclude, overrides)\u001b[0m\n\u001b[1;32m   2152\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mnot\u001b[39;00m (path \u001b[39m/\u001b[39m \u001b[39m\"\u001b[39m\u001b[39mvocab\u001b[39m\u001b[39m\"\u001b[39m)\u001b[39m.\u001b[39mexists() \u001b[39mand\u001b[39;00m \u001b[39m\"\u001b[39m\u001b[39mvocab\u001b[39m\u001b[39m\"\u001b[39m \u001b[39mnot\u001b[39;00m \u001b[39min\u001b[39;00m exclude:  \u001b[39m# type: ignore[operator]\u001b[39;00m\n\u001b[1;32m   2153\u001b[0m     \u001b[39m# Convert to list here in case exclude is (default) tuple\u001b[39;00m\n\u001b[1;32m   2154\u001b[0m     exclude \u001b[39m=\u001b[39m \u001b[39mlist\u001b[39m(exclude) \u001b[39m+\u001b[39m [\u001b[39m\"\u001b[39m\u001b[39mvocab\u001b[39m\u001b[39m\"\u001b[39m]\n\u001b[0;32m-> 2155\u001b[0m util\u001b[39m.\u001b[39;49mfrom_disk(path, deserializers, exclude)  \u001b[39m# type: ignore[arg-type]\u001b[39;00m\n\u001b[1;32m   2156\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_path \u001b[39m=\u001b[39m path  \u001b[39m# type: ignore[assignment]\u001b[39;00m\n\u001b[1;32m   2157\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_link_components()\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/spacy/util.py:1392\u001b[0m, in \u001b[0;36mfrom_disk\u001b[0;34m(path, readers, exclude)\u001b[0m\n\u001b[1;32m   1389\u001b[0m \u001b[39mfor\u001b[39;00m key, reader \u001b[39min\u001b[39;00m readers\u001b[39m.\u001b[39mitems():\n\u001b[1;32m   1390\u001b[0m     \u001b[39m# Split to support file names like meta.json\u001b[39;00m\n\u001b[1;32m   1391\u001b[0m     \u001b[39mif\u001b[39;00m key\u001b[39m.\u001b[39msplit(\u001b[39m\"\u001b[39m\u001b[39m.\u001b[39m\u001b[39m\"\u001b[39m)[\u001b[39m0\u001b[39m] \u001b[39mnot\u001b[39;00m \u001b[39min\u001b[39;00m exclude:\n\u001b[0;32m-> 1392\u001b[0m         reader(path \u001b[39m/\u001b[39;49m key)\n\u001b[1;32m   1393\u001b[0m \u001b[39mreturn\u001b[39;00m path\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/spacy/language.py:2131\u001b[0m, in \u001b[0;36mLanguage.from_disk.<locals>.deserialize_vocab\u001b[0;34m(path)\u001b[0m\n\u001b[1;32m   2129\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mdeserialize_vocab\u001b[39m(path: Path) \u001b[39m-\u001b[39m\u001b[39m>\u001b[39m \u001b[39mNone\u001b[39;00m:\n\u001b[1;32m   2130\u001b[0m     \u001b[39mif\u001b[39;00m path\u001b[39m.\u001b[39mexists():\n\u001b[0;32m-> 2131\u001b[0m         \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mvocab\u001b[39m.\u001b[39;49mfrom_disk(path, exclude\u001b[39m=\u001b[39;49mexclude)\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/spacy/vocab.pyx:504\u001b[0m, in \u001b[0;36mspacy.vocab.Vocab.from_disk\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/spacy/vectors.pyx:647\u001b[0m, in \u001b[0;36mspacy.vectors.Vectors.from_disk\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/spacy/util.py:1383\u001b[0m, in \u001b[0;36mfrom_disk\u001b[0;34m(path, readers, exclude)\u001b[0m\n\u001b[1;32m   1379\u001b[0m             writer(path \u001b[39m/\u001b[39m key)\n\u001b[1;32m   1380\u001b[0m     \u001b[39mreturn\u001b[39;00m path\n\u001b[0;32m-> 1383\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mfrom_disk\u001b[39m(\n\u001b[1;32m   1384\u001b[0m     path: Union[\u001b[39mstr\u001b[39m, Path],\n\u001b[1;32m   1385\u001b[0m     readers: Dict[\u001b[39mstr\u001b[39m, Callable[[Path], \u001b[39mNone\u001b[39;00m]],\n\u001b[1;32m   1386\u001b[0m     exclude: Iterable[\u001b[39mstr\u001b[39m],\n\u001b[1;32m   1387\u001b[0m ) \u001b[39m-\u001b[39m\u001b[39m>\u001b[39m Path:\n\u001b[1;32m   1388\u001b[0m     path \u001b[39m=\u001b[39m ensure_path(path)\n\u001b[1;32m   1389\u001b[0m     \u001b[39mfor\u001b[39;00m key, reader \u001b[39min\u001b[39;00m readers\u001b[39m.\u001b[39mitems():\n\u001b[1;32m   1390\u001b[0m         \u001b[39m# Split to support file names like meta.json\u001b[39;00m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# Get DD_Range, CC_Range, and HPI_Range for each file in txt_df_subset and store in tuples\n",
    "DD_Range = txt_df_subset['DD_Range'].iloc[0]\n",
    "CC_Range = txt_df_subset['CC_Range'].iloc[0]\n",
    "HPI_Range = txt_df_subset['HPI_Range'].iloc[0]\n",
    "\n",
    "# Create a dict that loops through each line in the txt_df_subset 'text' column in the DD_Range and stores each diagnosis as a key and the value as an embedding of the diagnosis using get_embedding from openai.\n",
    "nlp = spacy.load(\"en_core_sci_lg\")\n",
    "nlp.add_pipe(\"scispacy_linker\", config={\"resolve_abbreviations\": True, \"linker_name\": \"umls\"})\n",
    "diagnosis_dict = {}\n",
    "for i, line in enumerate(str(txt_df_subset['text'].iloc[0][DD_Range[0]:DD_Range[1]]).split('\\n')):\n",
    "    if i == 0:\n",
    "        continue\n",
    "    if len(line) > 0:\n",
    "        doc = nlp(line)\n",
    "        linker = nlp.get_pipe(\"scispacy_linker\")\n",
    "        for entity in doc.ents:\n",
    "            print(entity)\n",
    "            diagnosis_dict[entity] = []\n",
    "            for umls_ent in entity._.kb_ents:\n",
    "                try:\n",
    "                    for ent in umls_ent:\n",
    "                        print(ent)\n",
    "                    object = linker.kb.cui_to_entity[umls_ent[0]]\n",
    "                    for alias in object[2]:\n",
    "                        diagnosis_dict[entity].append(alias)\n",
    "                except Exception as e:\n",
    "                    print(e)\n",
    "\n",
    "print(diagnosis_dict)\n",
    "count_dict = {}\n",
    "for key, _ in diagnosis_dict.items():\n",
    "    count_dict[key] = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Chief Complaint', 'Primary Complaint', 'Presenting complaint', '{Chief complaint}', 'Chief complaint', 'Presenting Complaint', 'Chief complaint (finding)', 'Main Complain', 'chief complaint']\n",
      "['Other Chief Complaint']\n",
      "['Chief complaint duration', 'Duration of Chief Complaint']\n",
      "[]\n",
      "[]\n",
      "['Post Cardiac Arrest Syndrome', 'Postresuscitation Disease', 'Postresuscitation Diseases', 'Post-Cardiac Arrest Syndromes', 'Postcardiac Arrest Syndromes', 'Postcardiac Arrest Syndrome']\n",
      "['arrest [as an cardiac arrest]', 'Cardiac arrest (disorder)', 'ARREST CARDIAC', 'Cardiac arrest- asystole', 'arrest', 'cardiac arrest', 'CARDIAC ASYSTOLE', 'ASYSTOLIA', 'Asystole (disorder)', 'ventricular asystole', 'Asystole', 'VENTRICULAR ASYSTOLIA', 'cardiac asystole', 'Heart Arrest', 'Arrest, Heart', 'asystolia', 'Arrest, Cardiac', 'SCA', 'Cardiac Arrest', 'Cardiac arrest', 'asystole', 'Heart stops beating', 'heart arrest', 'Asystoles', 'STANDSTILL CARDIAC', 'Cardiac standstill', 'ASYSTOLE', 'HEART ARREST', 'arrested', 'CARDIAC ARREST']\n",
      "['acute asthma exacerbation', 'exacerbation of asthma', 'of asthma exacerbation', 'exacerbation asthma', 'Exacerbation of asthma (disorder)', 'Acute asthma', 'asthma exacerbation', 'asthma exacerbations', 'Exacerbation of asthma', 'Asthma attack', 'acute exacerbation of asthma']\n",
      "['status asthmaticus', 'asthma asthmaticus status', 'asthmaticus status', 'Asthma, unspecified type, with status asthmaticus', 'STATUS ASTHMATICUS', 'Asthmatic Crises', 'Shock, Asthmatic', 'Status Asthmaticus', 'Asthmatic Shocks', 'Shocks, Asthmatic', 'Asthmatic Crisis', 'Status asthmaticus', 'Asthma with status asthmaticus', 'Asthmatic Shock', 'severe acute asthma', 'acute severe asthma', 'Acute severe exacerbation of asthma', 'Asthmaticus, Status', 'Acute severe exacerbation of asthma (disorder)', 'Crisis, Asthmatic', 'asthmatic crisis', 'Crises, Asthmatic', 'Acute severe asthma']\n",
      "[]\n",
      "['ho', 'hos', 'History of present illness', 'HISTORY OF PRESENT ILLNESS', 'hpi', 'history']\n",
      "['Hx of Pres illness', 'Present illness:History:To identify measures at a point in time:^Patient:Nominal', 'Present illness:Hx:Pt:^Patient:Nom', 'History of Present illness']\n",
      "['year', 'Year', 'Annum', 'years', 'yr', 'year (qualifier value)']\n",
      "['year']\n",
      "['year']\n",
      "['/year', 'per year', 'Per Year', 'per year (qualifier value)', 'per years', 'Yearly']\n",
      "[]\n",
      "['asthmatics', 'asthmatic', 'ASTHMA', 'Bronchial asthma, NOS', 'Asthma', 'Asthma, Bronchial', 'Airway hyperreactivity', 'ASTHMA BRONCHIAL', 'ASTHMA, BRONCHIAL', 'Asthma, unspecified', 'asthma disorders', 'br asthma', 'Asthmas', 'Bronchial Asthma', 'Asthmatic', 'Bronchial asthma', 'asthma', 'Asthma (disorder)', 'Asthma, NOS', 'BRONCHIAL ASTHMA', 'bronchial asthma', 'bronchitic asthma']\n",
      "['status asthmaticus', 'asthma asthmaticus status', 'asthmaticus status', 'Asthma, unspecified type, with status asthmaticus', 'STATUS ASTHMATICUS', 'Asthmatic Crises', 'Shock, Asthmatic', 'Status Asthmaticus', 'Asthmatic Shocks', 'Shocks, Asthmatic', 'Asthmatic Crisis', 'Status asthmaticus', 'Asthma with status asthmaticus', 'Asthmatic Shock', 'severe acute asthma', 'acute severe asthma', 'Acute severe exacerbation of asthma', 'Asthmaticus, Status', 'Acute severe exacerbation of asthma (disorder)', 'Crisis, Asthmatic', 'asthmatic crisis', 'Crises, Asthmatic', 'Acute severe asthma']\n",
      "['Systemic hypertension', 'systemic hypertension', 'Hypertensive disorder', 'Arterial hypertension', 'Hypertension', 'Hypertensive vascular degeneration', 'Vascular Hypertensive Disorder', 'Hypertensive disease, NOS', 'hypertension', 'Blood Pressures, High', 'Hypertension, arterial', 'HT - Hypertension', 'hypertensive disorder', 'Blood Pressure, High', 'Hypertension (high blood pressure)', 'Hypertension, NOS', 'high bp', 'BLOOD PRESSURE HIGH', 'HTN - Hypertension', 'HYPERTENSION ARTERIAL', 'high blood pressure disorder', 'systemic arterial hypertension', 'BP+ - Hypertension', 'HYPERTENSION', 'Hyperpiesis', 'Hypertensive vascular disease', 'HBP', 'High blood pressure disorder', 'htn', 'Elevated blood pressure', 'hbp', 'Hypertensive cardiovascular disease or syndrome', 'BP - High blood pressure', 'hyperpiesis', 'vascular hypertension', 'HBP - High blood pressure', 'arterial hypertension', 'high blood pressure', 'vascular hypertensive disorder', 'arterial hypertension systemic', 'HTN', 'Hyperpiesia', 'hyperpiesia', 'Hypertensive disorder, systemic arterial (disorder)', 'High Blood Pressures', 'HYPERTENSIVE DISEASES', 'High blood pressure', 'High Blood Pressure', 'Systemic arterial hypertension', 'hypertensive disease', 'Hypertensive disorder, systemic arterial', 'hypertensive vascular disease', 'blood high pressure', 'Complications affecting other specified body systems, not elsewhere classified, hypertension', 'HYPERTENSIVE DISEASE']\n",
      "['dia htn', 'Diastolic Hypertension', 'BLOOD PRESSURE INCREASE DIASTOLIC', 'HYPERTENSION DIASTOLIC', 'diastolic hypertension', 'dia hypertension', 'DIASTOLIC HYPERTENSION', 'diastolic htn', 'Diastolic hypertension (disorder)', 'Hypertension, diastolic', 'Diastolic hypertension, NOS']\n",
      "['Hypertension, systolic', 'systolic htn', 'Systolic hypertension (disorder)', 'sys hypertension', 'SYSTOLIC HYPERTENSION', 'Systolic Hypertension', 'Systolic Hypertensions', 'Hypertension, Systolic', 'systolic hypertension', 'sys htn', 'Systolic hypertension']\n",
      "['Clinic Hypertension, Isolated', 'Hypertension, Isolated Clinic', 'labile hypertension', 'white coat hypertension', 'coat hypertension white', 'White Coat Hypertension', 'Isolated Clinic Hypertension', 'labile htn', 'White Coat Syndrome', 'Hypertension, White Coat', 'Labile hypertension due to being in a clinical environment', 'Syndrome, White Coat', 'Labile hypertension due to being in a clinical environment (disorder)', 'White coat hypertension', 'hypertension white coat']\n",
      "['CHTN']\n",
      "['Admission to hospital', 'hospital admission', 'Hospital admission, NOS', 'HOSPITAL ADMISSIONS', 'admit', 'Hospital Admission', 'Admission', 'Admit', 'admitted hospital', 'admission to hospital', 'Hospital admission', 'Hospital Admissions', 'admission hospital', 'hospital admissions', 'admissions hospital', 'admissions', 'admission', 'Hospital admission (procedure)', 'admitted', 'admits', 'admitting']\n",
      "['Admission', 'Admitted']\n",
      "[]\n",
      "['DOSH', '10-deoxysarpagine,NADPH:oxygen oxidoreductase (10-hydroxylating) activity']\n",
      "['Dyspnea, NOS', 'BREATHLESSNESS', 'Difficulty breathing', 'Breathless', 'Dyspnoea', 'SHORTNESS OF BREATH', 'difficult breathing', 'Breath Shortness', 'BREATHING DIFFICULT', 'Shortness of breath dyspnea', 'sob', 'breathing difficulty', 'breathing shortness', 'shortness of breath', 'DIFFICULTY BREATHING', 'DYSPNEA', 'SOB - Shortness of breath', 'Breathlessness', 'breath shortness', 'SOB', 'respiratory difficulty', 'Shortness of Breath', 'shortness of breath (SOB)', 'BREATH SHORTNESS', 'Breathing difficulties', 'Dyspnea (finding)', 'SOB (SHORTNESS OF BREATH)', 'dyspnoea', 'Trouble breathing', 'Respiration difficult', 'DIB - Difficulty in breathing', 'breathlessness', 'tightness of breath', 'DYSPNOEA', 'breathing difficulties', 'dyspnea', 'Shortness of breath', 'Abnormal breathing', 'Dyspnea', 's.o.b.', 'breathless', 'Respiratory difficulties', 'Dyspnoea, NOS', 'Breathing difficulty', 'Difficulty breathing (finding)', 'Difficult to breathe']\n",
      "['shortness of breath at rest', 'Rest Dyspnea', 'Shortness of breath at rest', 'Dyspneas, Rest', 'Dyspnea, Rest', 'Dyspnea at rest (finding)', 'shortness of breath rest', 'Dyspnoea at rest', 'Breathless - at rest', 'Dyspnea at rest', 'Dyspnea At Rest', 'Rest dyspnea', 'Dyspnea resting', 'breath rest shortness', 'Dyspnea^resting', 'Breathlessness at rest', 'SOBAR - Shortness of breath at rest']\n",
      "[]\n",
      "[]\n",
      "['Shortness of breathing upon physical activity', 'Dyspnea on exertion', 'exertional dyspnoea', 'Dyspnea on effort', 'Dyspnoea on exertion', 'exertional dyspnea', 'Breathlessness on exertion', 'SOBOE - Shortness of breath on exertion', 'Exertional dyspnoea', 'doe', 'DYSPNEA, EXERTIONAL', 'Exertional breathlessness', 'exertion dyspnea', 'SHORTNESS OF BREATH (ON EXERTION)', 'Short of breath on exertion', 'DOE', 'Dyspnea on exertion (finding)', 'dyspnea on exertion', 'shortness of breath on exertion', 'shortness of breath during exercise', 'Dyspnea Exertional', 'dyspnea on exertion (DOE)', 'dyspnea exertional', 'Exertional dyspnea', 'Dyspnoea on effort', 'Dyspnea, exertional', 'short of breath on exertion', 'DYSPNEA (ON EXERTION)']\n",
      "['Admission to hospital', 'hospital admission', 'Hospital admission, NOS', 'HOSPITAL ADMISSIONS', 'admit', 'Hospital Admission', 'Admission', 'Admit', 'admitted hospital', 'admission to hospital', 'Hospital admission', 'Hospital Admissions', 'admission hospital', 'hospital admissions', 'admissions hospital', 'admissions', 'admission', 'Hospital admission (procedure)', 'admitted', 'admits', 'admitting']\n",
      "['Admission', 'Admitted']\n",
      "[]\n",
      "['Peas', 'Pea (substance)', 'peas', 'Pea', 'pea']\n",
      "['electromechanical dissociation', 'Cardiac arrest with electromechanical dissociation', 'Electromechanical Dissociation by ECG Finding', 'Pulseless Electrical Activity', 'PEA - Pulseless electrical activity', 'emd', 'EMD', 'electromechanical dissociation (EMD)', 'Electro-Mechanical Dissociation', 'Electromechanical Dissociation by EKG Finding', 'Pulseless electrical activity', 'Electromechanical dissociation', 'PEA', 'pulseless electrical activity', 'Electromechanical dissociation (disorder)', 'Electromechanical Dissociation']\n",
      "['garden pea', 'peas', 'Pisum sativum', 'sativum, Pisum', 'pea', 'pisum sativum']\n",
      "['PEA']\n",
      "['Pisum sativum allergenic extract', 'PEA', 'Pea', 'pea allergenic extract']\n",
      "['arrest [as an cardiac arrest]', 'Cardiac arrest (disorder)', 'ARREST CARDIAC', 'Cardiac arrest- asystole', 'arrest', 'cardiac arrest', 'CARDIAC ASYSTOLE', 'ASYSTOLIA', 'Asystole (disorder)', 'ventricular asystole', 'Asystole', 'VENTRICULAR ASYSTOLIA', 'cardiac asystole', 'Heart Arrest', 'Arrest, Heart', 'asystolia', 'Arrest, Cardiac', 'SCA', 'Cardiac Arrest', 'Cardiac arrest', 'asystole', 'Heart stops beating', 'heart arrest', 'Asystoles', 'STANDSTILL CARDIAC', 'Cardiac standstill', 'ASYSTOLE', 'HEART ARREST', 'arrested', 'CARDIAC ARREST']\n",
      "['Arrested', 'Arrest', 'Arrested (qualifier value)', 'Arresting']\n",
      "['arrest', 'legal arrest', 'arrests', 'arresting']\n",
      "['Arrester']\n",
      "['S-Antigen, Retinal', 'Arrestin', 'Retinal S-Antigen', 'S Antigen, Retinal', 'Retinal S Antigen', 'arrestin']\n",
      "['patients', 'patient', 'Patient', 'Patient (person)', '*^patient', '^Patient', 'Patients']\n",
      "['Patient']\n",
      "['Patient']\n",
      "['Patient']\n",
      "['Patient']\n",
      "['PIM Kinase Inhibitor LGH447', 'LGH 447', 'LGH447', 'PIM447', 'PIM-447', 'PIM 447', 'LGH-447', 'PIM kinase inhibitor LGH447']\n",
      "['Male Impotence', 'Impotence, Male Sexual', 'Sexual Impotence, Male', 'male erectile disorder', 'Sexual impotence, NOS', 'dysfunctioning erectile', 'Difficulty getting an erection', 'Difficulty getting a full erection', 'Impotence', 'Erectile dysfunction (disorder)', 'erection dysfunction', 'Male Erectile Disorder', 'Erectile dysfunction', 'Erectile Dysfunction', 'Erectile abnormalities', 'Male erectile disorder', 'dysfunction erection', 'sexual impotence', 'erectile dysfunctions', 'Male Sexual Impotence', 'Failure of erection', 'impotent', 'dysfunction erectile', 'PENILE ERECTION IMPAIRMENT', 'impotence', 'Impotent', 'Abnormal erection', 'IMPOTENCE', 'Impotence, Male', 'Dysfunction, Erectile', 'Impotence, NOS', 'Sexual impotence', 'ED', 'erectile dysfunction']\n",
      "['Emergency (DEEDS)', 'Emergency Department (DEEDS) variables', 'ED']\n",
      "['hypoxaemia', 'Arterial hypoxemia', 'Low blood oxygen level', 'Blood oxygen decreased', 'Arterial hypoxaemia', 'Hypoxaemia', 'arterial hypoxemia', 'HYPOXEMIA', 'hypoxemia', 'arterial hypoxaemia', 'Hypoxemia (disorder)', 'hypoxemias', 'Hypoxemia', 'HYPOXAEMIA']\n",
      "[]\n",
      "[]\n",
      "['Proxemic']\n",
      "['RESPIRATORY DISTRESS', 'respiratory distress', 'Labored breathing', 'DISTRESS RESPIRATORY', 'Breathing difficulties', 'Respiratory difficulties', 'Respiratory Distress', 'Distressed respiration', 'Respiratory distress', 'Distressed breathing', 'distressed breathing', 'Respiratory distress (finding)', 'distressed respiratory']\n",
      "['acute respiratory distress', 'Acute Respiratory Distress', 'Acute respiratory distress (finding)', 'acute distress respiratory', 'Acute respiratory distress']\n",
      "['Perinatal respiratory distress, NOS']\n",
      "['Neonatal respiratory distress', 'Respiratory distress (newborn)', 'Infantile respiratory distress', 'distress respiratory neonatal', 'Respiratory distress, neonatal', 'neonatal respiratory distress', 'Neonatal respiratory distress (finding)', 'Newborn respiratory distress']\n",
      "['RESPIRATORY DISTRESS SYNDROME NEWBORN', 'distress respiratory syndrome', 'Idiopathic respiratory distress syndrome, NOS', 'infant respiratory distress syndrome', 'Hyaline membrane disease', 'Pulmonary hypoperfusion syndrome of newborn', 'Infant ARDS', 'Neonatal Respiratory Distress Syndrome', 'Newborn Respiratory Distress Syndrome', 'Respiratory Distress Syndrome, Infant', 'Congenital alveolar dysplasia', 'distress respiratory syndromes', 'Respiratory distress syndrome in the newborn (disorder)', 'newborn rds', 'Neonatal respiratory distress syndrome', 'RESPIRATORY DISTRESS SYNDROME, IDIOPATHIC', 'Respiratory distress syndrome of newborn', 'Cardiorespiratory distress syndrome of newborn', 'Wet lung disease of newborn', 'Respiratory distress syndrome in neonate', 'respiratory syndrome distress', 'Idiopathic respiratory distress syndrome of newborn', 'Respiratory distress syndrome in newborn', 'Infant acute respiratory distress syndrome', 'IRDS of newborn', 'SURFACTANT DEFICIENCY SYNDROME NEONATAL', 'Respiratory distress syndrome in the newborn', 'Idiopathic respiratory distress syndrome [IRDS or RDS] of newborn', 'RESPIRATORY DISTRESS SYNDROME, NEONATAL', 'Respiratory Distress Syndrome, Newborn', 'Idiopathic respiratory distress syndrome', 'Infant respiratory distress syndrome', 'Respiratory Distress Syndrome', 'DISTRESS RESPIRATORY SYNDROME NEWBORN', 'idiopathic respiratory distress of newborn', 'IRDS - Idiopathic respiratory distress syndrome', 'Infantile Respiratory Distress Syndrome', 'newborns rds', 'respiratory distress syndrome of newborn', 'RDS of newborn', 'SYNDROME RESPIRATORY DISTRESS NEWBORN', 'RDS, type I', 'RESPIRATORY DISTRESS SYNDROME, PERINATAL', 'respiratory distress syndrome', 'RESP DISTRESS SYNDROME NEONATAL']\n",
      "['DOSH', '10-deoxysarpagine,NADPH:oxygen oxidoreductase (10-hydroxylating) activity']\n",
      "['cyclophosphane', 'cyts', 'N,N-bis(2-chloroethyl)tetrahydro-2H-1,3,2-oxazaphosphorin-2-amine 2-Oxide', '(RS)-Cyclophosphamide', '(+-)-Cyclophosphamide', 'Cyclophosphamide-containing product', 'Mitoxan', 'Cyclophospham', 'Cyclophosphane', 'Cyclophosphanum', 'CTX', 'Cyclophosphan', 'cyclophosphamide', '2H-1,3,2-oxazaphosphorin-2-amine, N,N-is(2-chloroethyl)tetrahydro-,2-oxide, monohydrate', 'Cyclophosphamide (substance)', 'CYT', 'Cyclophosphamide', '2-[Bis(2-chloroethylamino)]-tetrahydro-2H-1,3,2-oxazaphosphorine-2-oxide', 'Claphene', 'cyt', 'CPM', 'Ciclofosfamide', 'N,N-Bis(2-chloroethyl)tetrahydro-2H-1,3,2-oxazaphosphorin-2-amine 2-oxide', 'Cyclophosphamidum', 'Cytophosphane', '(-)-cyclophosphamide', 'Cyclophosphamid', '2H-1,3,2-oxazaphosphorin-2-amine, N,N-is(2-chloroethyl)tetrahydro-,2-oxide', 'Bis(2-chloroethyl)phosphoramide cyclic propanolamide ester', 'CYCLOPHOSPHAMIDE', 'Product containing cyclophosphamide (medicinal product)', 'Ciclofosfamida', 'N,N-Bis(2-chloroethyl)tetrahydro-2H-1,3,2-oxazaphosphorin-2-amine 2-Oxide', 'Syklofosfamid']\n",
      "['Cerebrotendinous Xanthomatoses', 'Van Bogaert Scherer Epstein Disease', 'Cerebral cholesterinosis', 'Xanthomatoses, Cerebrotendinous', 'Cerebrotendinous Xanthomatosis', 'Van Bogaert-Scherer-Epstein Disease', 'CTX - Cerebrotendinous xanthomatosis', 'Cholestanolosis', 'CTX', 'CEREBROTENDINOUS XANTHOMATOSIS', 'Van Bogaert-Scherer-Epstein disease', 'CEREBRAL CHOLESTERINOSIS', 'Cerebrotendinous xanthomatosis syndrome', 'VON BOGAERT DISEASE', 'Bogaert-Scherer-Epstein Disease, Van', 'Cerebrotendinous cholesterinosis', 'Cerebrotendinous xanthomatosis', 'Cholestanol storage disease (disorder)', 'cerebrotendinous xanthomatosis', 'Cerebral Cholesterinoses', 'Disease, Van Bogaert-Scherer-Epstein', 'van Bogaert-Scherer-Epstein syndrome', 'Cerebral Cholesterinosis', 'Cholestanol storage disease', 'Xanthomatosis, Cerebrotendinous', 'Sterol 27-hydroxylase deficiency']\n",
      "['CP27', 'cerebrotendinous xanthomatosis', 'CTX', 'CYP27A1 Gene', 'CYP27', 'cytochrome P450 family 27 subfamily A member 1', 'CYP27A1', 'CYTOCHROME P450, SUBFAMILY XXVIIA, POLYPEPTIDE 1', 'cytochrome P450, family 27, subfamily A, polypeptide 1', 'CYP27A1 gene', 'STEROL 27-HYDROXYLASE', 'cytochrome P450, subfamily XXVIIA (steroid 27-hydroxylase, cerebrotendinous xanthomatosis), polypeptide 1', 'Cytochrome P450, Family 27, Subfamily A, Polypeptide 1 Gene']\n",
      "['CP27', 'CTX', 'CYP27', 'Cytochrome P450, Family 27, Subfamily A, Polypeptide 1 wt Allele', 'CYP27A1 wt Allele']\n",
      "['C-Terminal Telopeptide Type 1 Collagen', 'C-Telopeptide', 'CTX', 'Collagen Type 1 Cross-linked C-Telopeptide', 'Cross-Linked C-Terminal Telopeptide of Type I Collagen', 'C-Terminal Telopeptide Type I Collagen', 'CTX-I', 'CTXI']\n",
      "['Product containing azithromycin (medicinal product)', 'Azithromycin', '(2R,3S,4R,5R,8R,10R,11R,12S,13S,14R)-13-[(2,6-Dideoxy-3-C-methyl-3-O-methyl-alpha-L-ribo-hexopyranosyl)oxy]-2-ethyl-3,4,10-trihydroxy-3,5,6,8,10,12,14-heptamethyl-11-[(3,4,6-trideoxy-3-(dimethylamino)-beta-D-xylo-hexopyranosyl]oxy]-1-oxa-6-azacyclopentade', 'Azithromycin (substance)', '(2R,3S,4R,5R,8R,10R,11R,12S,13S,14R)-13-[(2,6-Dideoxy-3-C-methyl-3-O-methyl-alpha-L-ribo-hexopyranosyl)oxy]-2-ethyl-3,4,10-trihydroxy-3,5,6,8,10,12,14-heptamethyl-11-[(3,4,6-trideoxy-3-(dimethylamino)-beta-D-xylo-hexopyranosyl)oxy]-1-oxa-6-azacyclopentade', 'Azithromycine', 'AZITHROMYCIN', 'Azithromycin-containing product', '1-Oxa-6-azacyclopentadecan-15-one, 13-((2,6-dideoxy-3-C-methyl-3-O-methyl-alpha-L-ribo-hexopyranosyl)oxy)-2-ethyl-3,4,10-trihydroxy-3,5,6,8,10,12,14-heptamethyl-11-((3,4,6-trideoxy-3-(dimethylamino)-beta-D-xylo-hexopyranosyl)oxy)-, (2R-(2R*,3S*,4R*,5R*,8R*,10R*,11R*,12S*,13S*,14R*))-', 'N-methyl-11-aza-10-deoxy-10-dihydroerythromycin A', 'Azithromycinum', 'azythromycin', 'azithromycin', 'Azitromicina', 'Azythromycin']\n",
      "['azithromycin dihydrate', 'Azithromycin dihydrate', 'Azithromycin dihydrate (substance)', 'Azithromycin Dihydrate', 'AZITHROMYCIN DIHYDRATE', 'Dihydrate, Azithromycin']\n",
      "['South Carolina', 'SC', 'South Carolina (geographic location)', 'south carolina', 'carolina south']\n",
      "['SC', 'BLOOD GROUP--SCIANNA SYSTEM', 'Scianna blood group system (qualifier value)', 'ISBT system 013', 'ISBT system SC', 'SCIANNA BLOOD GROUP', 'Scianna blood group system']\n",
      "['SC', 'ERYTHROID MEMBRANE-ASSOCIATED PROTEIN', 'ERMAP', 'Radin blood group', 'erythroblast membrane associated protein (Scianna blood group)', 'erythroblast membrane-associated protein (Scianna blood group)', 'ERYTHROCYTE MEMBRANE-ASSOCIATED PROTEIN', 'ERMAP gene', 'erythroblast membrane-associated protein (RD and SC blood groups)', 'Scianna blood group', 'RD', 'BTN5', 'ERYTHROBLAST MEMBRANE-ASSOCIATED PROTEIN', 'erythroblast membrane-associated protein']\n",
      "['subcutaneous drug administration', 'Subcutaneous Route of Administration', 'SubcutaneousRoute', 'Subcutaneous route', 'Subcutaneous route (qualifier value)', 'SC use', 'Subcutaneous (intended site)', 'SC route', 'Subcutaneous use', 'Subcutaneous', 'Subdermal route']\n",
      "['(-)-(R)-epinephrine', 'Product containing epinephrine (medicinal product)', '4-(1-Hydroxy-2-(methylamino)ethyl)-1,2-benzenediol', 'Epinephrinum', 'Therapeutic Epinephrine', 'epi', '(R)-(-)-adrenaline', 'therapeutic epinephrine', 'Adrenaline preparation', '(R)-(-)-Adnephrine', 'Adrenaline', 'Epinephrine', 'epinephrine', '(-)-3,4-dihydroxy-α-((methylamino)methyl)benzyl alcohol', '1,2-Benzenediol, 4-(1-hydroxy-2-(methylamino)ethyl)-, (R)-', 'Adrenaline-containing product', '(R)-(-)-Epinephrine', '(R)-(-)-Adrenaline', '(R)-(-)-Epirenamine', 'adrenalin', 'Adrenalin', 'Epinefrin', 'EPINEPHrine', 'adrenaline', 'EPINEPHRINE', 'Epinephrine-containing product', 'Adrénaline', '4-[(1R)-1-Hydroxy-2-(methylamino)ethyl]-1,2-benzenediol', 'Epinefrina', 'Epinephrine (substance)', 'Epinephrine preparation', 'Epinephrin']\n",
      "['adrenaline binding']\n",
      "['Epinephrine racemic', 'EPINEPHRINE,RACEMIC', 'Adrenaline, Racemic', 'epinephrine racemic', 'Epinephrine, Racemic Mixture', 'Racemic Adrenaline', 'Racemic Epinephrine', 'dl-Epinephrine', 'Epinephrine, Racemic', 'racepinephrine', 'Racemic Mixture Adrenaline', 'racemic epinephrine', 'Racepinephrine (substance)', 'dl-Adrenaline', 'Racemic Mixture Epinephrine', 'RACEPINEPHRINE', 'Racepinephrine', 'Adrenaline, Racemic Mixture', 'Adrenaline racemic']\n",
      "['epinephrine formation', 'epinephrine anabolism', 'adrenaline biosynthesis', 'epinephrine synthesis', 'epinephrine biosynthesis', 'adrenaline biosynthetic process']\n",
      "['Epinephrine measurement', 'Adrenaline measurement', 'Epinephrine Measurement', 'Epinephrine measurement, NOS', 'Epinephrine measurement (procedure)', 'Adrenaline level', 'Adrenaline Measurement', 'ephinephrine level test', 'Epinephrine level']\n",
      "['solu medrol', 'solumedrol', 'Solu-Medrol', 'SOLU-Medrol', 'Solumedrol', 'solu-medrol']\n",
      "['DOSH', '10-deoxysarpagine,NADPH:oxygen oxidoreductase (10-hydroxylating) activity']\n",
      "['Episode', 'Episode (attribute)', 'episode', 'Episodes', 'Episode of', 'Episodes (qualifier value)', 'episodes']\n",
      "['Episode, Patient Care', 'Patient Care Episode', 'Episode of care', 'Care Episodes', 'Episodes, Patient Care', 'Episodes of Care', 'Care Episode', 'Patient Care Episodes', 'episodes of care', 'episode of care']\n",
      "['Recent', 'Recent (qualifier value)', 'Recent episode (qualifier value)', 'recent', 'Recent episode', 'Recently']\n",
      "['Episodic', 'Now and then', 'Episodic (qualifier value)', 'Recurrent']\n",
      "['Peas', 'Pea (substance)', 'peas', 'Pea', 'pea']\n",
      "['electromechanical dissociation', 'Cardiac arrest with electromechanical dissociation', 'Electromechanical Dissociation by ECG Finding', 'Pulseless Electrical Activity', 'PEA - Pulseless electrical activity', 'emd', 'EMD', 'electromechanical dissociation (EMD)', 'Electro-Mechanical Dissociation', 'Electromechanical Dissociation by EKG Finding', 'Pulseless electrical activity', 'Electromechanical dissociation', 'PEA', 'pulseless electrical activity', 'Electromechanical dissociation (disorder)', 'Electromechanical Dissociation']\n",
      "['garden pea', 'peas', 'Pisum sativum', 'sativum, Pisum', 'pea', 'pisum sativum']\n",
      "['PEA']\n",
      "['Pisum sativum allergenic extract', 'PEA', 'Pea', 'pea allergenic extract']\n",
      "['arrest [as an cardiac arrest]', 'Cardiac arrest (disorder)', 'ARREST CARDIAC', 'Cardiac arrest- asystole', 'arrest', 'cardiac arrest', 'CARDIAC ASYSTOLE', 'ASYSTOLIA', 'Asystole (disorder)', 'ventricular asystole', 'Asystole', 'VENTRICULAR ASYSTOLIA', 'cardiac asystole', 'Heart Arrest', 'Arrest, Heart', 'asystolia', 'Arrest, Cardiac', 'SCA', 'Cardiac Arrest', 'Cardiac arrest', 'asystole', 'Heart stops beating', 'heart arrest', 'Asystoles', 'STANDSTILL CARDIAC', 'Cardiac standstill', 'ASYSTOLE', 'HEART ARREST', 'arrested', 'CARDIAC ARREST']\n",
      "['Arrested', 'Arrest', 'Arrested (qualifier value)', 'Arresting']\n",
      "['arrest', 'legal arrest', 'arrests', 'arresting']\n",
      "['Arrester']\n",
      "['S-Antigen, Retinal', 'Arrestin', 'Retinal S-Antigen', 'S Antigen, Retinal', 'Retinal S Antigen', 'arrestin']\n",
      "[]\n",
      "['intubation', 'Intubation', 'Intubation (procedure)', 'Intubations', 'intubation procedures', 'intubation procedure', 'INTUBATION', 'Intubation Procedure', 'Intubation, NOS', 'intubations', 'Intubation - action', 'Intubation - action (qualifier value)']\n",
      "['(-)-(R)-epinephrine', 'Product containing epinephrine (medicinal product)', '4-(1-Hydroxy-2-(methylamino)ethyl)-1,2-benzenediol', 'Epinephrinum', 'Therapeutic Epinephrine', 'epi', '(R)-(-)-adrenaline', 'therapeutic epinephrine', 'Adrenaline preparation', '(R)-(-)-Adnephrine', 'Adrenaline', 'Epinephrine', 'epinephrine', '(-)-3,4-dihydroxy-α-((methylamino)methyl)benzyl alcohol', '1,2-Benzenediol, 4-(1-hydroxy-2-(methylamino)ethyl)-, (R)-', 'Adrenaline-containing product', '(R)-(-)-Epinephrine', '(R)-(-)-Adrenaline', '(R)-(-)-Epirenamine', 'adrenalin', 'Adrenalin', 'Epinefrin', 'EPINEPHrine', 'adrenaline', 'EPINEPHRINE', 'Epinephrine-containing product', 'Adrénaline', '4-[(1R)-1-Hydroxy-2-(methylamino)ethyl]-1,2-benzenediol', 'Epinefrina', 'Epinephrine (substance)', 'Epinephrine preparation', 'Epinephrin']\n",
      "['adrenaline binding']\n",
      "['Epinephrine racemic', 'EPINEPHRINE,RACEMIC', 'Adrenaline, Racemic', 'epinephrine racemic', 'Epinephrine, Racemic Mixture', 'Racemic Adrenaline', 'Racemic Epinephrine', 'dl-Epinephrine', 'Epinephrine, Racemic', 'racepinephrine', 'Racemic Mixture Adrenaline', 'racemic epinephrine', 'Racepinephrine (substance)', 'dl-Adrenaline', 'Racemic Mixture Epinephrine', 'RACEPINEPHRINE', 'Racepinephrine', 'Adrenaline, Racemic Mixture', 'Adrenaline racemic']\n",
      "['epinephrine formation', 'epinephrine anabolism', 'adrenaline biosynthesis', 'epinephrine synthesis', 'epinephrine biosynthesis', 'adrenaline biosynthetic process']\n",
      "['Epinephrine measurement', 'Adrenaline measurement', 'Epinephrine Measurement', 'Epinephrine measurement, NOS', 'Epinephrine measurement (procedure)', 'Adrenaline level', 'Adrenaline Measurement', 'ephinephrine level test', 'Epinephrine level']\n",
      "['endo-(+/-)-alpha-(Hydroxymethyl)benzeneacetic Acid 8-methyl-8-azabicyclo[3.2.1]oct-3-yl Ester', '(±)-atropine', 'atropine', 'Atropine', 'Atropinum', 'Atropina', 'Benzeneacetic acid, alpha-(hydroxymethyl)- 8-methyl-8-azabicyclo(3.2.1)oct-3-yl ester endo-(+-)-', '(±)-hyoscyamine', 'Atropine-containing product', 'dl-Tropyl Tropate', 'dl-tropyltropate', 'dl-Hyoscyamine', 'Atropin', 'Atropine (substance)', 'ATROPINE', 'Tropine tropate', 'Product containing atropine (medicinal product)', 'atropine [chemical substance]', 'Dl-hyoscyamine']\n",
      "['atropine [drug]', 'atropine']\n",
      "['Atropines']\n",
      "['atropine anabolism', 'atropine formation', 'atropine synthesis', 'atropine biosynthesis']\n",
      "['Product containing homatropine (medicinal product)', 'HOMATROPINE', 'Omatropina', 'Homatropine (substance)', 'homatropine', '(3-endo)-8-Methyl-8-azabicyclo(3.2.1)oct-3-yl hydroxy(phenyl)acetate', 'Homatropine-containing product', 'Homatropina', 'Homatropine']\n",
      "['Magnésium', 'mgs', 'Magnesium', 'MAGNESIUM', 'mg', 'Elemental Magnesium', 'Magnesio', 'Magnesium-containing product', 'magnesium', 'Magnesium, NOS', 'Mg', 'Magnesium (substance)', 'Magnesium metallicum', 'Mg element', 'Product containing magnesium (medicinal product)', 'Mg - Magnesium']\n",
      "['Magnesium Measurement', 'Magnesium', 'magnesium measurement', 'magnesium level test', 'Magnesium level', 'Magnesium measurement', 'Magnesium measurement, NOS', 'Magnesium measurement (procedure)']\n",
      "['Magnesium', 'Magnesium supplements, alimentary tract and metabolism']\n",
      "['MAGNESIUM']\n",
      "['Mg2+', 'Magnesium cation', 'Mg++ element', 'magnesium cation', 'Mn 2+', 'Magnesium Cation', 'MAGNESIUM CATION', 'Magnesium ion', 'magnesium ion', 'magnesium ions', 'Magnesium, ion(Mg2+)']\n",
      "['bicarbonate of soda', 'Sodium bicarbonate (substance)', 'Baking soda', 'sodium hydrogen carbonate', 'Sodium Bicarbonate', 'Sodium hydrogencarbonate', 'Bicarbonate, Sodium', 'Sodium acid carbonate', 'Bicarbonate of soda', 'Carbonic acid monosodium salt', 'Bicarbonate of Soda', 'Product containing sodium bicarbonate (medicinal product)', 'Soda, Baking', 'Sodium bicarbonate-containing product', 'Sodium Hydrogen Carbonate', 'SODIUM BICARBONATE', 'baking soda', 'Sodium bicarbonate', 'bicarb', 'Sodio bicarbonato', 'sodium bicarbonate', 'bicarbs', 'Hydrogen Carbonate, Sodium', 'Sodium hydrogen carbonate', 'Baking Soda', 'NaHCO3', 'Carbonic Acid Monosodium Salt', 'Natriumhydrogenkarbonat', 'bicarbonates sodium']\n",
      "['In addition to', 'Addition', 'Plus', 'W', 'PL', 'Cum', 'With (attribute)', 'With']\n",
      "['Addition', 'Add']\n",
      "['Add', 'Added', 'Additional']\n",
      "['Additive', 'additive']\n",
      "[]\n",
      "['Both', 'Right and Left', 'Both sides', 'Bilateral', 'bilateral', 'BILATERAL', 'Right and left', 'both sides', 'Right and left (qualifier value)']\n",
      "['Bilateral Cancer', 'Bilateral Malignant Tumor', 'Bilateral Malignant Neoplasm']\n",
      "['Bilateral Carcinoma', 'Bilateral Cancer']\n",
      "[]\n",
      "['Thoracostomies, Needle', 'Thoracostomy, Needle', 'Needle Thoracostomies']\n",
      "['Thoracostomies', 'thoracostomy', 'Thoracostomy', 'Thoracostomy, NOS']\n",
      "['report [document]', 'report', 'Report (record artifact)', 'reports', 'Reports', 'Report', 'document report', 'Reported']\n",
      "['informing', 'inform', 'Reporting', 'giving information', 'report', 'Informing (procedure)', 'Informing', 'reports', 'reported', 'reporting', 'Giving information', 'Reported']\n",
      "['Report', 'Report:Find:Pt:{Setting}:Doc:{Role}', 'Report:Finding:To identify measures at a point in time:{Setting}:Document:{Role}']\n",
      "['case reports', 'case report', 'case reported', 'case report [paper report]', 'Case Reports']\n",
      "['Case Report', 'Case Studies', 'case study', 'Case report', 'Case Study', 'case studies']\n",
      "['Air', 'Air (substance)', 'airs', 'Air, NOS', 'air']\n",
      "['AI/RHEUM', 'AIR']\n",
      "['AIR', 'ACUTE INSULIN RESPONSE']\n",
      "['IGF2R antisense RNA 1 (non-protein coding)', 'ANTISENSE IGF2R RNA, NONCODING', 'NCRNA00088', 'AIR', 'AIRN gene', 'AIRN', 'IGF2R-AS1', 'antisense of IGF2R non-protein coding RNA', 'non-protein coding RNA 88', 'antisense of IGF2R RNA (non-protein coding)', 'ANTISENSE IGF2R RNA', 'IGF2RAS']\n",
      "['Air', 'Air medicinal', 'AIR', 'Air, medicinal', 'medical air', 'Medical air', 'Medical air (medicinal product)', 'air']\n",
      "['Lt - Left', 'Left side', 'left sided', 'left', 'left sides', 'Left', 'Left (qualifier value)', 'Left - Side', 'left side', 'Levo-']\n",
      "['Left sided', 'Left-sided', 'left sided', 'Left sided (qualifier value)', 'Left', 'Left-Sided']\n",
      "['left']\n",
      "['Lower leg Left', 'Structure of left lower leg (body structure)', 'Leg Left', 'left legs', 'left leg', 'Structure of left lower leg', 'leg left', 'Left Leg', 'Left lower leg', 'Lower extremity.left>Lower leg', 'Leg-lower-left', 'Leg.left', 'lower left leg', 'Left leg']\n",
      "['Both', 'Right and Left', 'Both sides', 'Bilateral', 'bilateral', 'BILATERAL', 'Right and left', 'both sides', 'Right and left (qualifier value)']\n",
      "['Bilateral Cancer', 'Bilateral Malignant Tumor', 'Bilateral Malignant Neoplasm']\n",
      "['Bilateral Carcinoma', 'Bilateral Cancer']\n",
      "[]\n",
      "['Minute', 'min', 'minute of time', 'Min', 'minute', 'minimum', 'minutes']\n",
      "['Smaller', 'smaller', 'small', 'smalling', 'littles', 'Small', 'minute', 'little', 'Little', 'Small (qualifier value)', 'minutes']\n",
      "['Minute', 'Minuscule (qualifier value)', 'Minuscule', 'minute', 'minutes']\n",
      "['Minute', 'min', 'Arcminute', 'minute (qualifier value)', 'Minute Unit of Plane Angle', 'minute', 'minutes']\n",
      "['per minute', '/min', 'minutes per', 'per minute (qualifier value)', '/minute', 'Per Minute']\n",
      "['vecuronium', 'Vecuronio', 'Piperidinium, 1-((2Beta,3alpha,5alpha,16beta,17beta)-3,17-bis(acetyloxy)-2-(1-piperidinyl)androstan-16-yl)-1-methyl-', 'Vecuronium (substance)', 'VECURONIUM', 'Vecuronium-containing product', 'Vecuronium', 'Product containing vecuronium (medicinal product)']\n",
      "['VECURONIUM BROMIDE', 'Bromide, Vecuronium', 'vecuronium bromide', 'Piperidinium, 1-((2beta,3alpha,5alpha,16beta,17beta)-3,17-bis(acetyloxy)-2-(1-piperidinyl)androstan-16-yl)-1-methyl-, bromide', 'Piperidinium, 1-((2beta,3alpha,5alpha,16beta,17beta)-3,17-bis(acetyloxy)-2-(1-piperidinyl)androstan-16-yl)-1-methyl-, Bromide', 'Vecuronium Hydrobromide', 'bromide vecuronium', 'Vecuronium bromide (substance)', 'Vecuronium Bromide', 'Vecuronium bromide', 'piperidinium, 1-((2beta,3alpha,5alpha,16beta,17beta)-3,17-bis(acetyloxy)-2-(1-piperidinyl)androstan-16-yl)-1-methyl-, bromide', 'Hydrobromide, Vecuronium']\n",
      "['OGTT - Oral glucose tolerance test', 'glucose procedure testing tolerance', 'glucose challenge test', 'Glucose challenge test', 'glucose tolerance tests', 'GTT - Glucose tolerance test', 'Glucose tolerance test (procedure)', 'Glucose tolerance test, NOS', 'challenge glucose testing', 'glucose tests tolerance', 'glucose tolerance test', 'Glucose Tolerance Test', 'Glucose tolerance tests', 'gtts', 'Glucose tolerance test', 'gtt', 'glucose test tolerance', 'glucose testing tolerance', 'Glucose Tolerance Tests']\n",
      "['Gestational trophoblastic neoplasia (morphologic abnormality)', 'Persistent trophoblastic disease', 'Neoplasia, Gestational Trophoblastic', 'Gestational Trophoblastic Neoplasms', 'trophoblastic tumor, gestational', 'Gestational trophoblastic neoplasia', 'Gestational trophoblastic neoplasm', 'Trophoblastic Neoplasm, Gestational', 'Gestational Trophoblastic Tumor', 'Neoplasms, Gestational Trophoblastic', 'gestational trophoblastic tumor', 'trophoblastic tumor GTT', 'Gestational trophoblastic tumor', 'Gestational trophoblastic neoplasia (disorder)', 'Gestational Trophoblastic Neoplasm', 'Neoplasm, Gestational Trophoblastic', 'Trophoblastic Neoplasia, Gestational', 'GTN', 'Persistent gestational trophoblastic disease', 'GTT', 'Gestational Trophoblastic Neoplasia', 'Trophoblastic Neoplasms, Gestational']\n",
      "['asthmatics', 'asthmatic', 'ASTHMA', 'Bronchial asthma, NOS', 'Asthma', 'Asthma, Bronchial', 'Airway hyperreactivity', 'ASTHMA BRONCHIAL', 'ASTHMA, BRONCHIAL', 'Asthma, unspecified', 'asthma disorders', 'br asthma', 'Asthmas', 'Bronchial Asthma', 'Asthmatic', 'Bronchial asthma', 'asthma', 'Asthma (disorder)', 'Asthma, NOS', 'BRONCHIAL ASTHMA', 'bronchial asthma', 'bronchitic asthma']\n",
      "['status asthmaticus', 'asthma asthmaticus status', 'asthmaticus status', 'Asthma, unspecified type, with status asthmaticus', 'STATUS ASTHMATICUS', 'Asthmatic Crises', 'Shock, Asthmatic', 'Status Asthmaticus', 'Asthmatic Shocks', 'Shocks, Asthmatic', 'Asthmatic Crisis', 'Status asthmaticus', 'Asthma with status asthmaticus', 'Asthmatic Shock', 'severe acute asthma', 'acute severe asthma', 'Acute severe exacerbation of asthma', 'Asthmaticus, Status', 'Acute severe exacerbation of asthma (disorder)', 'Crisis, Asthmatic', 'asthmatic crisis', 'Crises, Asthmatic', 'Acute severe asthma']\n",
      "['cooling', 'Cool', 'Cooling']\n",
      "['Body cooling', 'body cooling', 'Active cooling of patient (procedure)', 'body cool']\n",
      "['cooler', 'cooling unit', 'coolers', 'Cooling Units', 'Coolers']\n",
      "['Cooling System', 'Cooling Module']\n",
      "['Hypothermia, Gastric', 'Gastric Lavages, Cold', 'Gastric freezing (procedure)', 'Gastric freezing', 'Lavage, Cold Gastric', 'Gastric cooling therapy', 'Cryotherapy of stomach', 'Cold Gastric Lavages', 'Gastric Lavage, Cold', 'Gastric cooling', 'Cold Gastric Lavage', 'Gastric cooling therapy (procedure)', 'Lavages, Cold Gastric', 'Gastric hypothermia']\n",
      "['Protocols (qualifier value)', 'Protocols', 'Protocol', 'protocol', 'protocols', 'protocols documentation']\n",
      "['DOC.REF.CTP', 'Clinical trial protocol document', 'Clinical Study Protocol', 'Clinical Trial Protocol reference document', 'Protocol', 'Study Protocol']\n",
      "['Library Protocol', 'Protocol']\n",
      "['Study Protocol', 'Protocol']\n",
      "['Protocol']\n",
      "['Surgical transfer - action', 'transfer', 'Transfer', 'surgical transfer', 'Transfer procedure', 'Surgical transfer - action (qualifier value)', 'transfered', 'Transfer - action', 'transferred']\n",
      "['Transfer', 'Delivery', 'Transport', 'Transferred', 'Deliver', 'Transfer Technique', 'Delivered']\n",
      "['patients transferred', 'patient transferring', 'Transfers, Patient', 'Patient Transfers', 'Patient transfer', 'PATIENT TRANSFERS', 'Patient Transfer', 'patient transfers', 'patients transfers', 'Transfer, Patient', 'Transfer care', 'Transfer of patient', 'patients transferring', 'patient transfer', 'Patient transfer (procedure)', 'patients transfer', 'Transfer of patient procedure']\n",
      "['Transfer']\n",
      "['ActClassTransfer', 'transfer']\n",
      "['Evaluated', 'efficacy assessment', 'evaluation', 'evaluations', 'Investigation', 'Evaluation', 'Evaluate', 'effectiveness assessment']\n",
      "['assessment', 'Evaluation - action', 'Evaluation procedure', 'Assessment', 'clinical investigation', 'evaluation', 'assessment procedure', 'Evaluation procedure (procedure)', 'Evaluation - action (qualifier value)', 'Investigations', 'Investigation', 'Evaluation', 'Patient evaluation - action', 'Assessment Procedures', 'assessments', 'Clinical evaluation']\n",
      "['Evaluation']\n",
      "['examinations mouth', 'oral examination', 'Oral examination (procedure)', 'oral examinations', 'Oral examination', 'Oral Examination', 'Examinations, Oral', 'examinations oral', 'Mouth--Examination', 'Examination of mouth', 'mouth examination', 'Oral Evaluation', 'Examination, Oral', 'Oral Examinations']\n",
      "['evaluation/testing', 'evaluation testing', 'evaluation tests', 'evaluation test', 'test evaluation']\n",
      "['patients', 'patient', 'Patient', 'Patient (person)', '*^patient', '^Patient', 'Patients']\n",
      "['Patient']\n",
      "['Patient']\n",
      "['Patient']\n",
      "['Patient']\n",
      "['Admission to hospital', 'hospital admission', 'Hospital admission, NOS', 'HOSPITAL ADMISSIONS', 'admit', 'Hospital Admission', 'Admission', 'Admit', 'admitted hospital', 'admission to hospital', 'Hospital admission', 'Hospital Admissions', 'admission hospital', 'hospital admissions', 'admissions hospital', 'admissions', 'admission', 'Hospital admission (procedure)', 'admitted', 'admits', 'admitting']\n",
      "['Admission', 'Admitted']\n",
      "[]\n",
      "['PIM Kinase Inhibitor LGH447', 'LGH 447', 'LGH447', 'PIM447', 'PIM-447', 'PIM 447', 'LGH-447', 'PIM kinase inhibitor LGH447']\n",
      "['Dyspnea, NOS', 'BREATHLESSNESS', 'Difficulty breathing', 'Breathless', 'Dyspnoea', 'SHORTNESS OF BREATH', 'difficult breathing', 'Breath Shortness', 'BREATHING DIFFICULT', 'Shortness of breath dyspnea', 'sob', 'breathing difficulty', 'breathing shortness', 'shortness of breath', 'DIFFICULTY BREATHING', 'DYSPNEA', 'SOB - Shortness of breath', 'Breathlessness', 'breath shortness', 'SOB', 'respiratory difficulty', 'Shortness of Breath', 'shortness of breath (SOB)', 'BREATH SHORTNESS', 'Breathing difficulties', 'Dyspnea (finding)', 'SOB (SHORTNESS OF BREATH)', 'dyspnoea', 'Trouble breathing', 'Respiration difficult', 'DIB - Difficulty in breathing', 'breathlessness', 'tightness of breath', 'DYSPNOEA', 'breathing difficulties', 'dyspnea', 'Shortness of breath', 'Abnormal breathing', 'Dyspnea', 's.o.b.', 'breathless', 'Respiratory difficulties', 'Dyspnoea, NOS', 'Breathing difficulty', 'Difficulty breathing (finding)', 'Difficult to breathe']\n",
      "['shortness of breath at rest', 'Rest Dyspnea', 'Shortness of breath at rest', 'Dyspneas, Rest', 'Dyspnea, Rest', 'Dyspnea at rest (finding)', 'shortness of breath rest', 'Dyspnoea at rest', 'Breathless - at rest', 'Dyspnea at rest', 'Dyspnea At Rest', 'Rest dyspnea', 'Dyspnea resting', 'breath rest shortness', 'Dyspnea^resting', 'Breathlessness at rest', 'SOBAR - Shortness of breath at rest']\n",
      "[]\n",
      "[]\n",
      "['Shortness of breathing upon physical activity', 'Dyspnea on exertion', 'exertional dyspnoea', 'Dyspnea on effort', 'Dyspnoea on exertion', 'exertional dyspnea', 'Breathlessness on exertion', 'SOBOE - Shortness of breath on exertion', 'Exertional dyspnoea', 'doe', 'DYSPNEA, EXERTIONAL', 'Exertional breathlessness', 'exertion dyspnea', 'SHORTNESS OF BREATH (ON EXERTION)', 'Short of breath on exertion', 'DOE', 'Dyspnea on exertion (finding)', 'dyspnea on exertion', 'shortness of breath on exertion', 'shortness of breath during exercise', 'Dyspnea Exertional', 'dyspnea on exertion (DOE)', 'dyspnea exertional', 'Exertional dyspnea', 'Dyspnoea on effort', 'Dyspnea, exertional', 'short of breath on exertion', 'DYSPNEA (ON EXERTION)']\n",
      "['diagnosing', 'Diagnosis (observable entity)', 'Diagnostic', 'diagnosis', 'diagnoses', 'diagnosed', '{Diagnosis}', 'diagnose', 'DIAGNOSIS', 'Diagnosed', 'diagnosis (DX)', 'Diagnosis', 'Diagnose', 'Diagnoses', 'Dx']\n",
      "[]\n",
      "['Missing Diagnosis', 'Diagnoses, Missed', 'Diagnoses, Missing', 'Missing Diagnoses', 'Diagnosis, Missing', 'Missed Diagnoses', 'Diagnosis, Missed']\n",
      "['diagnosis']\n",
      "['Diagnoses, Oral', 'Oral Diagnoses', 'Oral Diagnosis']\n",
      "['Cap', 'Cap (physical object)']\n",
      "['CTX/DOX/PRED', 'cyclophosphamide/doxorubicin/prednisone', 'CAP', 'Cyclophosphamide/Doxorubicin/Prednisone']\n",
      "['Cisplatin/Cyclophosphamide/Doxorubicin', 'CDDP/CTX/DOX', 'CAP-II', 'CAP-I', 'PAC', 'CISCA chemotherapy protocol', 'CAP', 'CISCA-II', 'cisplatin/cyclophosphamide/doxorubicin', 'cisplatin,cyclophosphamide and doxorubicin']\n",
      "['placental leucine aminopeptidase', 'LEUCYL-CYSTINYL AMINOPEPTIDASE', 'PLAP', 'INSULIN-RESPONSIVE AMINOPEPTIDASE, MOUSE, HOMOLOG OF', 'P-LAP', 'LNPEP gene', 'CAP', 'leucyl/cystinyl aminopeptidase', 'OXYTOCINASE', 'leucyl and cystinyl aminopeptidase', 'cystinyl aminopeptidase', 'LNPEP', 'LEUCINE AMINOPEPTIDASE OF PLACENTA', 'IRAP']\n",
      "['serpin peptidase inhibitor, clade B (ovalbumin), member 6', 'PI6', 'cytoplasmic antiproteinase', 'PROTEASE INHIBITOR 6', 'SERPINB6 gene', 'serpin family B member 6', 'SPI3', 'SERPINB6', 'deafness, autosomal recessive 91', 'CAP', 'PTI', 'serine (or cysteine) proteinase inhibitor, clade B (ovalbumin), member 6', 'placental thrombin inhibitor', 'DFNB91', 'SPI3, MOUSE, HOMOLOG OF', 'PLACENTAL THROMBIN INHIBITOR', 'SERPIN PEPTIDASE INHIBITOR, CLADE B (OVALBUMIN), MEMBER 6']\n",
      "['asthmatics', 'asthmatic', 'ASTHMA', 'Bronchial asthma, NOS', 'Asthma', 'Asthma, Bronchial', 'Airway hyperreactivity', 'ASTHMA BRONCHIAL', 'ASTHMA, BRONCHIAL', 'Asthma, unspecified', 'asthma disorders', 'br asthma', 'Asthmas', 'Bronchial Asthma', 'Asthmatic', 'Bronchial asthma', 'asthma', 'Asthma (disorder)', 'Asthma, NOS', 'BRONCHIAL ASTHMA', 'bronchial asthma', 'bronchitic asthma']\n",
      "['status asthmaticus', 'asthma asthmaticus status', 'asthmaticus status', 'Asthma, unspecified type, with status asthmaticus', 'STATUS ASTHMATICUS', 'Asthmatic Crises', 'Shock, Asthmatic', 'Status Asthmaticus', 'Asthmatic Shocks', 'Shocks, Asthmatic', 'Asthmatic Crisis', 'Status asthmaticus', 'Asthma with status asthmaticus', 'Asthmatic Shock', 'severe acute asthma', 'acute severe asthma', 'Acute severe exacerbation of asthma', 'Asthmaticus, Status', 'Acute severe exacerbation of asthma (disorder)', 'Crisis, Asthmatic', 'asthmatic crisis', 'Crises, Asthmatic', 'Acute severe asthma']\n",
      "['Treated by', 'Treated with', 'TW', 'Treated with (attribute)']\n",
      "['Treat', 'Treatment', 'treated with', 'treating', 'Treated', 'Treating', 'treated']\n",
      "['cyclophosphane', 'cyts', 'N,N-bis(2-chloroethyl)tetrahydro-2H-1,3,2-oxazaphosphorin-2-amine 2-Oxide', '(RS)-Cyclophosphamide', '(+-)-Cyclophosphamide', 'Cyclophosphamide-containing product', 'Mitoxan', 'Cyclophospham', 'Cyclophosphane', 'Cyclophosphanum', 'CTX', 'Cyclophosphan', 'cyclophosphamide', '2H-1,3,2-oxazaphosphorin-2-amine, N,N-is(2-chloroethyl)tetrahydro-,2-oxide, monohydrate', 'Cyclophosphamide (substance)', 'CYT', 'Cyclophosphamide', '2-[Bis(2-chloroethylamino)]-tetrahydro-2H-1,3,2-oxazaphosphorine-2-oxide', 'Claphene', 'cyt', 'CPM', 'Ciclofosfamide', 'N,N-Bis(2-chloroethyl)tetrahydro-2H-1,3,2-oxazaphosphorin-2-amine 2-oxide', 'Cyclophosphamidum', 'Cytophosphane', '(-)-cyclophosphamide', 'Cyclophosphamid', '2H-1,3,2-oxazaphosphorin-2-amine, N,N-is(2-chloroethyl)tetrahydro-,2-oxide', 'Bis(2-chloroethyl)phosphoramide cyclic propanolamide ester', 'CYCLOPHOSPHAMIDE', 'Product containing cyclophosphamide (medicinal product)', 'Ciclofosfamida', 'N,N-Bis(2-chloroethyl)tetrahydro-2H-1,3,2-oxazaphosphorin-2-amine 2-Oxide', 'Syklofosfamid']\n",
      "['Cerebrotendinous Xanthomatoses', 'Van Bogaert Scherer Epstein Disease', 'Cerebral cholesterinosis', 'Xanthomatoses, Cerebrotendinous', 'Cerebrotendinous Xanthomatosis', 'Van Bogaert-Scherer-Epstein Disease', 'CTX - Cerebrotendinous xanthomatosis', 'Cholestanolosis', 'CTX', 'CEREBROTENDINOUS XANTHOMATOSIS', 'Van Bogaert-Scherer-Epstein disease', 'CEREBRAL CHOLESTERINOSIS', 'Cerebrotendinous xanthomatosis syndrome', 'VON BOGAERT DISEASE', 'Bogaert-Scherer-Epstein Disease, Van', 'Cerebrotendinous cholesterinosis', 'Cerebrotendinous xanthomatosis', 'Cholestanol storage disease (disorder)', 'cerebrotendinous xanthomatosis', 'Cerebral Cholesterinoses', 'Disease, Van Bogaert-Scherer-Epstein', 'van Bogaert-Scherer-Epstein syndrome', 'Cerebral Cholesterinosis', 'Cholestanol storage disease', 'Xanthomatosis, Cerebrotendinous', 'Sterol 27-hydroxylase deficiency']\n",
      "['CP27', 'cerebrotendinous xanthomatosis', 'CTX', 'CYP27A1 Gene', 'CYP27', 'cytochrome P450 family 27 subfamily A member 1', 'CYP27A1', 'CYTOCHROME P450, SUBFAMILY XXVIIA, POLYPEPTIDE 1', 'cytochrome P450, family 27, subfamily A, polypeptide 1', 'CYP27A1 gene', 'STEROL 27-HYDROXYLASE', 'cytochrome P450, subfamily XXVIIA (steroid 27-hydroxylase, cerebrotendinous xanthomatosis), polypeptide 1', 'Cytochrome P450, Family 27, Subfamily A, Polypeptide 1 Gene']\n",
      "['CP27', 'CTX', 'CYP27', 'Cytochrome P450, Family 27, Subfamily A, Polypeptide 1 wt Allele', 'CYP27A1 wt Allele']\n",
      "['C-Terminal Telopeptide Type 1 Collagen', 'C-Telopeptide', 'CTX', 'Collagen Type 1 Cross-linked C-Telopeptide', 'Cross-Linked C-Terminal Telopeptide of Type I Collagen', 'C-Terminal Telopeptide Type I Collagen', 'CTX-I', 'CTXI']\n",
      "['Product containing azithromycin (medicinal product)', 'Azithromycin', '(2R,3S,4R,5R,8R,10R,11R,12S,13S,14R)-13-[(2,6-Dideoxy-3-C-methyl-3-O-methyl-alpha-L-ribo-hexopyranosyl)oxy]-2-ethyl-3,4,10-trihydroxy-3,5,6,8,10,12,14-heptamethyl-11-[(3,4,6-trideoxy-3-(dimethylamino)-beta-D-xylo-hexopyranosyl]oxy]-1-oxa-6-azacyclopentade', 'Azithromycin (substance)', '(2R,3S,4R,5R,8R,10R,11R,12S,13S,14R)-13-[(2,6-Dideoxy-3-C-methyl-3-O-methyl-alpha-L-ribo-hexopyranosyl)oxy]-2-ethyl-3,4,10-trihydroxy-3,5,6,8,10,12,14-heptamethyl-11-[(3,4,6-trideoxy-3-(dimethylamino)-beta-D-xylo-hexopyranosyl)oxy]-1-oxa-6-azacyclopentade', 'Azithromycine', 'AZITHROMYCIN', 'Azithromycin-containing product', '1-Oxa-6-azacyclopentadecan-15-one, 13-((2,6-dideoxy-3-C-methyl-3-O-methyl-alpha-L-ribo-hexopyranosyl)oxy)-2-ethyl-3,4,10-trihydroxy-3,5,6,8,10,12,14-heptamethyl-11-((3,4,6-trideoxy-3-(dimethylamino)-beta-D-xylo-hexopyranosyl)oxy)-, (2R-(2R*,3S*,4R*,5R*,8R*,10R*,11R*,12S*,13S*,14R*))-', 'N-methyl-11-aza-10-deoxy-10-dihydroerythromycin A', 'Azithromycinum', 'azythromycin', 'azithromycin', 'Azitromicina', 'Azythromycin']\n",
      "['azithromycin dihydrate', 'Azithromycin dihydrate', 'Azithromycin dihydrate (substance)', 'Azithromycin Dihydrate', 'AZITHROMYCIN DIHYDRATE', 'Dihydrate, Azithromycin']\n",
      "['Admission to hospital', 'hospital admission', 'Hospital admission, NOS', 'HOSPITAL ADMISSIONS', 'admit', 'Hospital Admission', 'Admission', 'Admit', 'admitted hospital', 'admission to hospital', 'Hospital admission', 'Hospital Admissions', 'admission hospital', 'hospital admissions', 'admissions hospital', 'admissions', 'admission', 'Hospital admission (procedure)', 'admitted', 'admits', 'admitting']\n",
      "['pre admissions', 'Pre-admission (qualifier value)', 'Pre-admission', 'pre-admission', 'pre admission']\n",
      "['Admission', 'Admitted']\n",
      "['Admissions, Patient', 'patient admissions', 'patient admission', 'client admission', 'admission patients', 'Admission, Patient', 'Patient Admissions']\n",
      "['WINTER', 'winter']\n",
      "['Yellow Rockets', 'upland cress', 'vulgari, Barbarea', 'Rocket, Yellow', 'Cress, Winter', 'Yellow rocket', 'Winter Cress', 'Barbarea vulgaris', 'Rockets, Yellow', 'Barbarea vulgaris (organism)', 'Cresses, Winter', 'Barbarea vulgari', 'yellow rocket', 'Yellow Rocket', 'Winter Cresses']\n",
      "['HFH11', 'PIG29', 'TGT3', 'Trident Gene', 'FKHL16', 'MPHOSPH2', 'Forkhead Box M1 wt Allele', 'FOXM1 wt Allele', 'MPP-2', 'MPP2', 'INS-1', 'FOXM1B', 'WIN', 'HNF-3', 'HFH-11', 'TRIDENT']\n",
      "['acute asthma exacerbation', 'exacerbation of asthma', 'of asthma exacerbation', 'exacerbation asthma', 'Exacerbation of asthma (disorder)', 'Acute asthma', 'asthma exacerbation', 'asthma exacerbations', 'Exacerbation of asthma', 'Asthma attack', 'acute exacerbation of asthma']\n",
      "['status asthmaticus', 'asthma asthmaticus status', 'asthmaticus status', 'Asthma, unspecified type, with status asthmaticus', 'STATUS ASTHMATICUS', 'Asthmatic Crises', 'Shock, Asthmatic', 'Status Asthmaticus', 'Asthmatic Shocks', 'Shocks, Asthmatic', 'Asthmatic Crisis', 'Status asthmaticus', 'Asthma with status asthmaticus', 'Asthmatic Shock', 'severe acute asthma', 'acute severe asthma', 'Acute severe exacerbation of asthma', 'Asthmaticus, Status', 'Acute severe exacerbation of asthma (disorder)', 'Crisis, Asthmatic', 'asthmatic crisis', 'Crises, Asthmatic', 'Acute severe asthma']\n",
      "[]\n",
      "['Male Impotence', 'Impotence, Male Sexual', 'Sexual Impotence, Male', 'male erectile disorder', 'Sexual impotence, NOS', 'dysfunctioning erectile', 'Difficulty getting an erection', 'Difficulty getting a full erection', 'Impotence', 'Erectile dysfunction (disorder)', 'erection dysfunction', 'Male Erectile Disorder', 'Erectile dysfunction', 'Erectile Dysfunction', 'Erectile abnormalities', 'Male erectile disorder', 'dysfunction erection', 'sexual impotence', 'erectile dysfunctions', 'Male Sexual Impotence', 'Failure of erection', 'impotent', 'dysfunction erectile', 'PENILE ERECTION IMPAIRMENT', 'impotence', 'Impotent', 'Abnormal erection', 'IMPOTENCE', 'Impotence, Male', 'Dysfunction, Erectile', 'Impotence, NOS', 'Sexual impotence', 'ED', 'erectile dysfunction']\n",
      "['Emergency (DEEDS)', 'Emergency Department (DEEDS) variables', 'ED']\n",
      "['arterial blood gase', 'Analysis of arterial blood gases and pH', 'Gases, arterial measurement', 'ABG - Arterial blood gas analysis', 'Analysis of arterial blood gases and pH (procedure)', 'Measurement of systemic arterial blood gases', 'Arterial blood gas analysis', 'analysis arterial blood gas', 'ABG', 'arterial blood gas', 'Blood gases, arterial measurement (procedure)', 'Arterial Blood Gas Measurement', 'Blood Gas Measurement', 'Analysis of arterial blood gas (oxygen saturation, pO2, pCO2, CO2, pH); rest only', 'Blood Gas Analysis', 'ABGs measurement', 'arterial blood gases', 'Arterial blood gases', 'Blood gases, arterial measurement', 'blood gas arterial', 'arterial blood gas analysis']\n",
      "['A1BG', 'ALPHA-1-B-GLYCOPROTEIN', 'GAB', 'ABG', 'alpha-1-B glycoprotein', 'A1BG gene', 'A1B']\n",
      "['Bypass, Aortocoronary', 'Coronary artery bypass grafting', 'coronary artery bypass graft', 'Vascular aortocoronary bypass', 'ACB graft', 'Aortocoronary bypass grafting', 'Coronary artery bypass graft', 'coronary bypass', 'Bypasses, Aortocoronary', 'Cardiac revascularization with bypass anastomosis', 'Cardiac revascularization with aortocoronary bypass anastomosis', 'Coronary artery bypass grafts (CABG)', 'Aortocoronary artery bypass graft (procedure)', 'Coronary bypass', 'Coronary Artery Bypass Graft', 'Coronary artery bypass', 'coronary artery bypass graft (GABG)', 'Bypass anastomosis for heart revascularization', 'Coronary Artery Bypass Grafting', 'CABG - Coronary artery bypass graft', 'CAB graft', 'Aortocoronary Bypasses', 'coronary bypass surgery', 'Coronary Artery Bypasses', 'Aortocoronary bypass for heart revascularization', 'Coronary artery bypass graft (CABG)', 'CORONARY ARTERY BYPASS', 'Cardiac revascularisation with bypass anastomosis', 'Coronary artery bypass grafting (procedure)', 'Cardiac revascularisation with aortocoronary bypass anastomosis', 'Artery Bypasses, Coronary', 'Aortocoronary artery bypass graft', 'Aortocoronary bypass for heart revascularization, NOS', 'Aorto-coronary artery bypass graft', 'CORONARY BYPASS', 'Artery Bypass, Coronary', 'Bypass, Coronary Artery', 'Cardiac revascularization with bypass anastomosis (procedure)', 'Aortocoronary Bypass', 'CBG - Coronary bypass graft', 'Coronary Artery Bypass Surgery', 'Bypass Surgery, Coronary Artery', 'Coronary artery bypass graft operations', 'Aortocoronary bypass for heart revascularisation', 'coronary artery bypass', 'Coronary artery bypass graft surgery', 'CABG', 'CAG - Coronary artery graft', 'Heart Bypass', 'Coronary Artery Bypass', 'coronary artery bypass surgery', 'Bypasses, Coronary Artery', 'aortocoronary bypass', 'Aortocoronary bypass for heart revascularization, not otherwise specified', 'Cardiac revascularization with bypass anastomosis, NOS']\n",
      "['CTH', 'CSE', 'cystathionase (cystathionine gamma-lyase)', 'CYSTATHIONASE', 'CYSTATHIONINE GAMMA-LYASE', 'cystathionine gamma-lyase', 'CTH gene']\n",
      "['CTH', 'VSIG2', 'CTX, XENOPUS, HOMOLOG-LIKE', 'CTXL', 'VSIG2 gene', 'V-set and immunoglobulin domain containing 2', 'V-SET AND IMMUNOGLOBULIN DOMAINS-CONTAINING PROTEIN 2']\n",
      "['Adrenocorticotrophin', 'adrenocorticotrophic hormone', 'CCTROP', 'Hormone, Adrenocorticotropic', 'Adrenocorticotropic hormone', 'corticotropin, repository', 'corticotrophins', 'Adrenocorticotropin', 'corticotropins', 'adrenocorticotropin', 'Corticotrophin (1-39)', 'Adrenocorticotrophic hormone', 'ACTH', '1-39 ACTH', 'Corticotrophin', 'Corticotropin', 'adrenocorticotropic peptide', 'Hormone, Adrenocorticotrophic', 'corticotrophin', 'Corticotropin preparation', 'acth hormone', 'Adrenocorticotropic hormone (substance)', 'Adrenocorticotrophic Hormone', 'adrenocorticotropic hormone (ACTH)', 'Corticotropin (1-39)', 'adrenotrophin', 'Adrenocorticotropic Hormone', 'CORTICOTROPIN', 'ACTH preparation', 'adrenotropin', 'ACTH - Adrenocorticotrophic hormone', 'ACTH (1-39)', 'corticotropin', 'adrenocorticotropic hormone', 'Adrenocorticotropic hormone preparation', 'Adrenocorticotropic hormone (ACTH)', 'adrenotropic hormone']\n",
      "['beta-endorphin', 'proopiomelanocortin', 'opiomelanocortin prepropeptide', 'LPH', 'NPP', 'adrenocorticotropin', 'PROOPIOMELANOCORTIN', 'ACTH', 'POMC Gene', 'MELANOCYTE-STIMULATING HORMONE', 'MSH', 'POMC gene', 'POC', 'Proopiomelanocortin (Adrenocorticotropin/ Beta-Lipotropin/ Alpha-Melanocyte Stimulating Hormone/ Beta-Melanocyte Stimulating Hormone/ Beta-Endorphin) Gene', 'PRO-ACTH-ENDORPHIN', 'MELANOTROPIN', 'POMC', 'LIPOTROPIN', 'beta-melanocyte stimulating hormone', 'alpha-melanocyte stimulating hormone', 'beta-lipotropin', 'adrenocorticotropic hormone', 'CLIP']\n",
      "['NPP', 'POC', 'POMC wt Allele', 'ACTH', 'MSH', 'Proopiomelanocortin (Adrenocorticotropin/ Beta-Lipotropin/ Alpha-Melanocyte Stimulating Hormone/ Beta-Melanocyte Stimulating Hormone/ Beta-Endorphin) wt Allele', 'CLIP', 'LPH']\n",
      "['Chest', 'regions thoracic', 'Thoraces', 'CHEST', 'thoracics', 'thorax', 'thoracic', 'Thorace', 'THORAX', 'area thoracic', 'Chests', 'Thoracic (qualifier value)', 'Thoracic structure', 'Thorax, NOS', 'chest', 'thoracic area', 'Thoracic', 'thoraces', 'Thorax', 'chest and upper back', 'Thoracic structure (body structure)', 'chest region', 'thoracic region']\n",
      "['Chest', 'Subdivision of chest', 'Anterior thoracic region', 'Front of thorax', 'Chest region', 'Anterolateral part of thorax']\n",
      "['Peas', 'Pea (substance)', 'peas', 'Pea', 'pea']\n",
      "['electromechanical dissociation', 'Cardiac arrest with electromechanical dissociation', 'Electromechanical Dissociation by ECG Finding', 'Pulseless Electrical Activity', 'PEA - Pulseless electrical activity', 'emd', 'EMD', 'electromechanical dissociation (EMD)', 'Electro-Mechanical Dissociation', 'Electromechanical Dissociation by EKG Finding', 'Pulseless electrical activity', 'Electromechanical dissociation', 'PEA', 'pulseless electrical activity', 'Electromechanical dissociation (disorder)', 'Electromechanical Dissociation']\n",
      "['garden pea', 'peas', 'Pisum sativum', 'sativum, Pisum', 'pea', 'pisum sativum']\n",
      "['PEA']\n",
      "['Pisum sativum allergenic extract', 'PEA', 'Pea', 'pea allergenic extract']\n",
      "['arrest [as an cardiac arrest]', 'Cardiac arrest (disorder)', 'ARREST CARDIAC', 'Cardiac arrest- asystole', 'arrest', 'cardiac arrest', 'CARDIAC ASYSTOLE', 'ASYSTOLIA', 'Asystole (disorder)', 'ventricular asystole', 'Asystole', 'VENTRICULAR ASYSTOLIA', 'cardiac asystole', 'Heart Arrest', 'Arrest, Heart', 'asystolia', 'Arrest, Cardiac', 'SCA', 'Cardiac Arrest', 'Cardiac arrest', 'asystole', 'Heart stops beating', 'heart arrest', 'Asystoles', 'STANDSTILL CARDIAC', 'Cardiac standstill', 'ASYSTOLE', 'HEART ARREST', 'arrested', 'CARDIAC ARREST']\n",
      "['Arrested', 'Arrest', 'Arrested (qualifier value)', 'Arresting']\n",
      "['arrest', 'legal arrest', 'arrests', 'arresting']\n",
      "['Arrester']\n",
      "['S-Antigen, Retinal', 'Arrestin', 'Retinal S-Antigen', 'S Antigen, Retinal', 'Retinal S Antigen', 'arrestin']\n",
      "['Minute', 'min', 'minute of time', 'Min', 'minute', 'minimum', 'minutes']\n",
      "['Smaller', 'smaller', 'small', 'smalling', 'littles', 'Small', 'minute', 'little', 'Little', 'Small (qualifier value)', 'minutes']\n",
      "['Minute', 'Minuscule (qualifier value)', 'Minuscule', 'minute', 'minutes']\n",
      "['Minute', 'min', 'Arcminute', 'minute (qualifier value)', 'Minute Unit of Plane Angle', 'minute', 'minutes']\n",
      "['per minute', '/min', 'minutes per', 'per minute (qualifier value)', '/minute', 'Per Minute']\n",
      "['(-)-(R)-epinephrine', 'Product containing epinephrine (medicinal product)', '4-(1-Hydroxy-2-(methylamino)ethyl)-1,2-benzenediol', 'Epinephrinum', 'Therapeutic Epinephrine', 'epi', '(R)-(-)-adrenaline', 'therapeutic epinephrine', 'Adrenaline preparation', '(R)-(-)-Adnephrine', 'Adrenaline', 'Epinephrine', 'epinephrine', '(-)-3,4-dihydroxy-α-((methylamino)methyl)benzyl alcohol', '1,2-Benzenediol, 4-(1-hydroxy-2-(methylamino)ethyl)-, (R)-', 'Adrenaline-containing product', '(R)-(-)-Epinephrine', '(R)-(-)-Adrenaline', '(R)-(-)-Epirenamine', 'adrenalin', 'Adrenalin', 'Epinefrin', 'EPINEPHrine', 'adrenaline', 'EPINEPHRINE', 'Epinephrine-containing product', 'Adrénaline', '4-[(1R)-1-Hydroxy-2-(methylamino)ethyl]-1,2-benzenediol', 'Epinefrina', 'Epinephrine (substance)', 'Epinephrine preparation', 'Epinephrin']\n",
      "['Magnetic Resonance Imaging, Echo Planar', 'Echo-Planar Imaging', 'Medical Imaging, Planar', 'Magnetic Resonance Imaging, Echo-Planar', 'Imaging, Echoplanar', 'Echoplanar Imaging', 'Imaging, Echo-Planar', 'Echo Planar Imaging', 'Magnetic Resonance Imaging, Echoplanar', 'Echoplanar Imagings', 'Echo-Planar Magnetic Resonance Imaging', 'echo-planar imaging', 'Echoplanar Magnetic Resonance Imaging', 'Imagings, Echoplanar', 'Imagings, Echo-Planar', 'Echo Planar Magnetic Resonance Imaging', 'EPI', 'echo planar imaging', 'Echo-Planar Imagings']\n",
      "['Pancreatic exocrine insufficiency', 'Insufficiencies, Exocrine Pancreatic', 'Insufficiency, Exocrine Pancreatic', 'Exocrine Pancreatic Insufficiency', 'Pancreatic Insufficiency', 'Exocrine Pancreas Insufficiency', 'Exocrine pancreatic insufficiency (disorder)', 'Exocrine pancreatic insufficiency (EPI)', 'Pancreatic Insufficiencies, Exocrine', 'Exocrine pancreatic insufficiency', 'Insufficiency, Pancreatic', 'Pancreatic Insufficiencies', 'Pancreatic Insufficiency, Exocrine', 'Exocrine Pancreatic Insufficiencies', 'EPI', 'Inability to properly digest food due to lack of pancreatic digestive enzymes', 'Insufficiencies, Pancreatic']\n",
      "['Eysenck Personality Inventory', 'EPI - Eysenck personality inventory', 'Eysenck personality inventory', 'EPI', 'eysenck personality inventory (EPI)', 'Eysenck personality inventory (assessment scale)']\n",
      "['TFPI Gene', 'tissue factor pathway inhibitor', 'TFPI', 'LACI', 'EXTRINSIC PATHWAY INHIBITOR', 'LIPOPROTEIN-ASSOCIATED COAGULATION INHIBITOR', 'EPI', 'TISSUE FACTOR PATHWAY INHIBITOR', 'tissue factor pathway inhibitor (lipoprotein-associated coagulation inhibitor)', 'TFPI gene', 'TFI', 'TFPI1', 'extrinsic pathway inhibitor', 'Tissue Factor Pathway Inhibitor (Lipoprotein-Associated Coagulation Inhibitor) Gene']\n",
      "['bicarbonate of soda', 'Sodium bicarbonate (substance)', 'Baking soda', 'sodium hydrogen carbonate', 'Sodium Bicarbonate', 'Sodium hydrogencarbonate', 'Bicarbonate, Sodium', 'Sodium acid carbonate', 'Bicarbonate of soda', 'Carbonic acid monosodium salt', 'Bicarbonate of Soda', 'Product containing sodium bicarbonate (medicinal product)', 'Soda, Baking', 'Sodium bicarbonate-containing product', 'Sodium Hydrogen Carbonate', 'SODIUM BICARBONATE', 'baking soda', 'Sodium bicarbonate', 'bicarb', 'Sodio bicarbonato', 'sodium bicarbonate', 'bicarbs', 'Hydrogen Carbonate, Sodium', 'Sodium hydrogen carbonate', 'Baking Soda', 'NaHCO3', 'Carbonic Acid Monosodium Salt', 'Natriumhydrogenkarbonat', 'bicarbonates sodium']\n",
      "['Admission to hospital', 'hospital admission', 'Hospital admission, NOS', 'HOSPITAL ADMISSIONS', 'admit', 'Hospital Admission', 'Admission', 'Admit', 'admitted hospital', 'admission to hospital', 'Hospital admission', 'Hospital Admissions', 'admission hospital', 'hospital admissions', 'admissions hospital', 'admissions', 'admission', 'Hospital admission (procedure)', 'admitted', 'admits', 'admitting']\n",
      "['Admission', 'Admitted']\n",
      "[]\n",
      "['administration [executive duty]', 'Administration, Management', 'Administration (navigational concept)', 'Management', 'Administration', 'administration']\n",
      "['Management, Disease', 'management', 'Disease management', 'disease management', 'Disease Managements', 'Managements, Disease', 'Disease Management']\n",
      "['Supervision', 'Management procedure', 'Managing', 'Oversight', 'management procedures', 'Management', 'management procedure', 'Managed', 'Management procedure (procedure)']\n",
      "['Management', 'Management Occupations']\n",
      "['Care management', 'care management']\n",
      "['patients', 'patient', 'Patient', 'Patient (person)', '*^patient', '^Patient', 'Patients']\n",
      "['Patient']\n",
      "['Patient']\n",
      "['Patient']\n",
      "['Patient']\n",
      "[]\n",
      "['intubation', 'Intubation', 'Intubation (procedure)', 'Intubations', 'intubation procedures', 'intubation procedure', 'INTUBATION', 'Intubation Procedure', 'Intubation, NOS', 'intubations', 'Intubation - action', 'Intubation - action (qualifier value)']\n",
      "['Under sedation', 'Sedated', 'sedation [status]', 'Sedated state', 'sedation', 'sedating', 'under sedation', 'sedates', 'sedate', 'Sedation', 'SEDATION', 'Sedated (finding)', 'sedated']\n",
      "['Sedation procedure', 'sedation', 'Administration of sedative (procedure)', 'sedation [procedure]', 'Sedative therapy', 'Administration of sedative', 'Sedation', 'Sedation procedure, NOS', 'Sedation, NOS']\n"
     ]
    }
   ],
   "source": [
    "for key, _ in diagnosis_dict.items():\n",
    "    count_dict[key] = 0\n",
    "curr_text = str(txt_df_subset['text'].iloc[0])\n",
    "# text_to_analyze = \"\"\n",
    "# # Add the text from the CC_Range and HPI_Range to text_to_analyze\n",
    "# text_to_analyze += curr_text[CC_Range[0]:CC_Range[1]]\n",
    "# text_to_analyze += curr_text[HPI_Range[0]:HPI_Range[1]]\n",
    "\n",
    "for line in nltk.sent_tokenize(curr_text):\n",
    "    doc = nlp(line)\n",
    "    linker = nlp.get_pipe(\"scispacy_linker\")\n",
    "    for entity in doc.ents:\n",
    "        for umls_ent in entity._.kb_ents:\n",
    "            try:\n",
    "                object = linker.kb.cui_to_entity[umls_ent[0]]\n",
    "                print(object[2])\n",
    "                for key, arr in diagnosis_dict.items():\n",
    "                    for alias in object[2]:\n",
    "                        if alias in arr:\n",
    "                            count_dict[key] += 1\n",
    "                            break\n",
    "            except Exception as e:\n",
    "                print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{Anoxic: [], Brain Injury: ['Brain Damage', 'cerebral injury', 'damage brain', 'brain injury tissue', 'brain injuries', 'brain lesion (from injury)', 'Brain damage, NOS', 'Cerebral damage', 'Brain Injuries', 'brain tissue injury', 'Brain damage', 'Brain Injury', 'injury brain', 'Injuries, Brain', 'cerebral damage', 'Acquired brain injury', 'disorder brain injury', 'Damage, brain', 'brain injury', 'Injury, Brain', 'brain disorders injury', 'brain damages', 'Brain injury', 'brain damage', 'Chronic Brain Injury', 'Chronic Brain Injuries', 'chronic brain injury', 'Brain Injuries, Chronic', 'Injury', 'Traumatic or non-traumatic injury (disorder)', 'Traumatic or non-traumatic injury', 'Trauma', 'Traumatic injury', 'Injury', 'injuries', 'Traumatic injury (disorder)', 'injury', 'Injury, NOS', 'Injury - disorder', 'TRAUMATIC INJURY', 'Injuries', 'INJURY', 'injury from an external force', 'Wound', 'Traumatic Injury'], PEA arrest x2: [], Status Asthmaticus: ['status asthmaticus', 'asthma asthmaticus status', 'asthmaticus status', 'Asthma, unspecified type, with status asthmaticus', 'STATUS ASTHMATICUS', 'Asthmatic Crises', 'Shock, Asthmatic', 'Status Asthmaticus', 'Asthmatic Shocks', 'Shocks, Asthmatic', 'Asthmatic Crisis', 'Status asthmaticus', 'Asthma with status asthmaticus', 'Asthmatic Shock', 'severe acute asthma', 'acute severe asthma', 'Acute severe exacerbation of asthma', 'Asthmaticus, Status', 'Acute severe exacerbation of asthma (disorder)', 'Crisis, Asthmatic', 'asthmatic crisis', 'Crises, Asthmatic', 'Acute severe asthma', 'Acute severe refractory exacerbation of asthma (disorder)', 'Status asthmaticus', 'Acute severe refractory exacerbation of asthma', 'Acute severe asthma', 'asthmatics', 'asthmatic', 'ASTHMA', 'Bronchial asthma, NOS', 'Asthma', 'Asthma, Bronchial', 'Airway hyperreactivity', 'ASTHMA BRONCHIAL', 'ASTHMA, BRONCHIAL', 'Asthma, unspecified', 'asthma disorders', 'br asthma', 'Asthmas', 'Bronchial Asthma', 'Asthmatic', 'Bronchial asthma', 'asthma', 'Asthma (disorder)', 'Asthma, NOS', 'BRONCHIAL ASTHMA', 'bronchial asthma', 'bronchitic asthma'], Ventilator: ['ventilators', 'ventilator', 'Ventilator equipment', 'Ventilators', 'Ventilator (physical object)', 'Ventilator equipment (physical object)', 'Ventilator', 'Ventilator', 'portable ventilators', 'care homes ventilator', 'Ventilators, Volume, Portable', 'portable ventilator', 'Household ventilator, device (physical object)', 'Home Care Ventilators', 'Household ventilator, device', 'Ventilators, Home Care, Portable', 'Ventilators', 'Portable Ventilators', 'Ventilators, Portable/Home Care', 'Ventilator', 'Ventilator care (regime/therapy)', 'Ventilator Care', 'ventilator care', 'care ventilator', 'ventilation', 'Ventilations', 'ventilations', 'environmental air flow', 'air ventilation', 'Ventilate', 'Ventilation'], Pneumonia: ['Lung inflamed', 'Pulmonitis', 'PNEUMONIA', 'Pneumonias', 'Pneumonia (disorder)', 'Pneumonia, NOS', 'Pneumonia', 'pneum', 'pulmonary inflammation', 'pneumonia', 'unspecified pneumonia', 'inflammation lungs', 'lung inflammation', 'Pulmonary inflammation', 'Pneumonitis, NOS', 'Pneumonias, Mycoplasma', 'Atypical Pneumonias, Primary', 'Primary Atypical Pneumonias', 'Pneumonia caused by Genus Mycoplasma (disorder)', 'Mycoplasma Pneumonias', 'Eaton agent pneumonia', 'Endemic pneumonia', 'mycoplasmas pneumonia', 'Mycoplasmal pneumonia', 'Pneumonia, Primary Atypical', 'Pneumonia caused by Mycoplasma pneumoniae', 'primary atypical pneumonia', \"Pneumonia due to Eaton's agent\", 'Mycoplasmal Pneumonia', 'cold agglutinin positive pneumonia', 'Pneumonias, Primary Atypical', 'Pneumonia due to mycoplasma pneumoniae', 'Pneumonia due to Mycoplasma pneumoniae', 'Primary atypical pneumonia due to Mycoplasma pneumoniae', 'mycoplasma pneumonia', 'PNEUMONIA, COLD AGGLUTININ POSITIVE', 'Pneumonia, Mycoplasma', 'PPLO', 'MYCOPLASMA PNEUMONIA', 'Mycoplasma Pneumonia', 'pneumonia mycoplasma', 'Atypical Pneumonia, Primary', 'Primary atypical pneumonia (PAP) due to mycoplasma', 'Mycoplasma pneumonia', 'Primary Atypical Pneumonia', 'Pneumonia caused by Mycoplasma pneumoniae (disorder)', 'Mycoplasma pneumoniae pneumonia', 'PNEUMONIA, EATON AGENT', 'Primary atypical pneumonia caused by Mycoplasma pneumoniae', 'PNEUMONIA, MYCOPLASMAL', 'pplo', 'mycoplasmal pneumonia', 'PAP due to Mycoplasma pneumoniae', \"Eaton's agent pneumonia\", 'PNEUMONIA, PRIMARY ATYPICAL', 'PCP Infections', 'P carinii Pneumonias', 'Pneumonia, Pneumocystis carinii', 'Pneumonia, Interstitial Plasma Cell', 'Pneumocystis pneumonia', 'PNEUMONIA, PNEUMOCYSTIS CARINII', 'PNEUMOCYSTIS CARINII PNEUMONIA', 'Pneumonia, Pneumocystis', 'Pneumocystosis (disorder)', 'Pneumonia, P. carinii', 'Pneumonia, P. jirovecii', 'pcp', 'Pneumocystis carinii pneumonia', 'Pneumocystosis jirovecii pneumonia', 'Pneumonias, PCP', 'PNEUMOCYSTIS PNEUMONIA', 'Pulmonary pneumocystosis', 'PNEUMONIA, PNEUMOCYSTIS', 'Pneumocystis jirovecii pneumonia', 'PCP Pneumonias', 'Pneumonia, PCP', 'interstitial plasma cell pneumonia', 'P. carinii Pneumonia', 'PCP - Pneumocystis pneumonia', 'PCP Infection', 'P. carinii Pneumonias', 'Infection, PCP', 'Pneumonia, Pneumocystis jirovecii', 'Pneumocystis', 'pneumocystis carinii pneumonia (PCP)', 'PNEUMONIA, INTERSTITIAL PLASMA CELL', 'pneumocystosis', 'Pneumocystosis, pulmonary', 'Pneumocystis carinii Pneumonia', 'P. jirovecii Pneumonias', 'Interstitial plasma cell pneumonia', 'P carinii Pneumonia', 'P. jirovecii Pneumonia', 'Pneumocystoses', 'Pneumocystis jirovecii Pneumonia', 'Pneumocystis Carinii Pneumonia', 'Pneumocystosis jirovecii pneumonia (disorder)', 'Pneumonia, P carinii', 'Pneumocystis Pneumonias', 'Pneumocystis Pneumonia', 'Pneumocystosis', 'pneumocystis pneumonia', 'Pneumocystosis pneumonia', 'pneumocystis carinii pneumonia', 'PCP', 'Pneumonia due to Pneumocystis carinii', 'PCP Pneumonia', 'Inflammation, Lung', 'inflammation lung', 'Inflammations, Pulmonary', 'Pulmonary Inflammation', 'Pneumonitis (disorder)', 'Lung Inflammations', 'Pneumonitides', 'Pulmonary Inflammations', 'Inflammations, Lung', 'Inflammation, Pulmonary', 'Lung Inflammation', 'PNEUMONITIS', 'Lung inflammation', 'Pulmonary inflammation', 'pulmonitis', 'pneumonias', 'pneumonitis', 'Pneumonitis', 'Pneumonia due to Streptococcus', 'Streptococcal Pneumonia', 'Lobar pneumonia', 'PNEUMONIA, STREPTOCOCCAL', 'Pneumococcal pneumonia', 'streptococcal pneumonia', 'PNEUMONIA, PNEUMOCOCCAL', 'PNEUMOCOCCAL PNEUMONIA', 'PNEUMONIA, HEMOLYTIC STREPTOCOCCAL', 'Pneumococcal pneumonia [Streptococcus pneumoniae pneumonia]', 'Streptococcus pneumoniae pneumonia', 'Pneumococcal pneumonia (disorder)', 'Pneumococcal Pneumonia', 'Streptococcal pneumonia', 'Streptococcus pneumonia', 'PNEUMONIA PNEUMOCOCCAL', 'Pneumonia caused by Streptococcus', 'Pneumonias, Pneumococcal', 'Pneumonia, Pneumococcal', 'Pneumonia caused by Streptococcus (disorder)', 'Pneumococcal Pneumonias', 'pneumonia streptococcal', 'Pneumonia due to Streptococcus, unspecified'], Chronic Systolic Heart Failure: ['Chronic systolic heart failure (disorder)', 'Chronic systolic heart failure', 'Heart Failures, Systolic', 'Heart Failure, Reduced Ejection Fraction', 'Heart Failure, Systolic', 'Systolic heart failure', 'Systolic Heart Failure', 'Systolic heart failure (disorder)', 'Systolic heart failure, unspecified', 'systolic heart failure', 'failure heart systolic', 'Systolic Heart Failures', 'heart failure systolic', 'Heart Failure, Normal Ejection Fraction', 'Diastolic Heart Failure', 'Diastolic heart failure, unspecified', 'Diastolic Heart Failures', 'Diastolic heart failure', 'Heart Failure, Preserved Ejection Fraction', 'heart failure diastolic', 'Diastolic heart failure (disorder)', 'diastolic heart failure', 'diastolic failure heart', 'Heart Failure, Diastolic'], L1 compression fracture: ['Fracture, Compression', 'Compression Fractures', 'Compression fracture (morphologic abnormality)', 'Compression Fracture', 'compression fracture', 'Compression fractures', 'Fractures, Compression', 'Compression fracture', 'Compression fracture (disorder)', 'Fracture, compression', 'Wedge fracture', 'compression fractures', 'compression fracture vertebra', 'vertebra compression fracture', 'fractured vertebra (compression)', 'Compression fractured vertebra'], Seizures: ['Seizure, NOS', 'Fit, NOS', 'Fits NOS', 'Convulsion', 'Fit', 'Seizure (finding)', 'fits', 'Convulsion, NOS', 'convulsions', 'CONVULSION', 'Seizure NOS', 'Epileptic seizure', 'seizure', 'Fitting', 'fit', 'SEIZURE', 'Seizures', 'sz', 'Seizure', 'seizures', 'CONVULSIONS', 'Fit - convulsion', 'CONVULSION (NOS)', 'fitted', 'convulsion', 'FITS NOS', \"fit's\", 'seizure d', 'Convulsions', 'convulsions nos', 'Crescendo seizure', 'Seizure flurries', 'Acute repetitive seizures', 'Recurrent seizure (disorder)', 'Serial seizure', 'Acute repetitive seizure (finding)', 'Cluster seizures', 'Recurrent seizures', 'Cluster seizures (finding)', 'Cluster seizure', 'Crescendo seizures', 'Serial seizures', 'Cyclical seizures', 'Seizure cluster', 'Recurrent seizure', 'local seizures', 'Seizure affecting one half of brain', 'Partial seizures', 'PARTIAL SEIZURES', 'seizure focal partial', 'Focal-onset seizure', 'Focal and Partial Seizures', 'focal seizure partial', 'focal fit', 'Focal seizure, NOS', 'FOCAL SEIZURE', 'Local convulsion', 'Partial Seizure', 'focal seizure', 'Seizure, Focal', 'Focal seizures', 'Seizure, Partial', 'Partial seizure (disorder)', 'partial focal seizure', 'partial focal seizures', 'focal seizures', 'seizures partial', 'Seizures, Focal', 'Partial seizures, NOS', 'Seizures, partial', 'Partial seizure', 'CONVULSIONS LOCAL', 'Focal Seizures', 'partial seizure', 'Focal seizure', 'Focal-onset seizures', 'Seizures, focal', 'Focal Seizure', 'Local seizure', 'Focal onset seizure', 'partial seizures', 'Seizures, partial, afebrile', 'Partial Seizures', 'Tonic Seizure', 'Tonic Seizures', 'convulsions tonic', 'tonic seizures', 'seizures tonic', 'Tonic seizure (finding)', 'Tonic seizures', 'Seizures, tonic', 'Tonic Convulsion', 'Seizure, Tonic', 'Seizures, Tonic', 'Tonic seizure', 'TONIC CONVULSION', 'tonic seizure', 'astatic seizures', 'Seizure, Atonic', 'Atonic seizures', 'Atonic seizure (disorder)', 'Hypotonic seizures', 'Atonic seizure', 'Hypotonic seizure', 'Drop attacks', 'Drop seizures', 'Atonic Seizures', 'atonic seizure', 'Atonic Seizure', 'Astatic seizures', 'Epileptic drop attack', 'Astatic seizure', 'atonic seizure disorder', 'astatic seizure', 'Seizures, atonic', 'atonic seizures', 'ATONIC SEIZURES', 'Epileptic drop attack (disorder)'], hypoxic: ['hypoxia', 'hypoxic', 'Hypoxic', 'oxygen deficiency', 'Hypoxia', 'Oxygen decrease', 'HYPOXIA', 'Deficiency, Oxygen', 'Decreased oxygen supply', 'Oxygen Deficiencies', 'Oxygen Deficiency', 'Hypoxia (disorder)', 'Deficiencies, Oxygen', 'HYPOXIA', 'Hypoxia, CTCAE', 'hypoxis', 'Hypoxis', 'hypoxi', 'Hypoxic Cell Sensitizer', 'Selective Hypoxic Cell Cytotoxic Agent', 'Hypoxic Cell Cytotoxic Agent', 'Hypoxia Selective Agent']}\n",
      "{Anoxic: 0, Brain Injury: 0, PEA arrest x2: 0, Status Asthmaticus: 8, Ventilator: 0, Pneumonia: 0, Chronic Systolic Heart Failure: 0, L1 compression fracture: 0, Seizures: 0, hypoxic: 0}\n",
      "Anoxic: Sign_symptom: 0\n",
      "\n",
      "Brain Injury: Disease_disorder: 0\n",
      "\n",
      "PEA arrest x2: Disease_disorder: 0\n",
      "\n",
      "Status Asthmaticus: Diagnostic_procedure: 8\n",
      "\n",
      "Ventilator: Therapeutic_procedure: 0\n",
      "\n",
      "Pneumonia: Disease_disorder: 0\n",
      "\n",
      "Chronic Systolic Heart Failure: Detailed_description: 0\n",
      "\n",
      "L1 compression fracture: Sign_symptom: 0\n",
      "\n",
      "Seizures: Sign_symptom: 0\n",
      "\n",
      "hypoxic: Sign_symptom: 0\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/codespace/.python/current/lib/python3.10/site-packages/transformers/pipelines/token_classification.py:169: UserWarning: `grouped_entities` is deprecated and will be removed in version v5.0.0, defaulted to `aggregation_strategy=\"simple\"` instead.\n",
      "  warnings.warn(\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:94: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  final_df = final_df.append(disease_df) # adding the disease_df to existing\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:100: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  master_df = master_df.append(final_df)\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/transformers/pipelines/token_classification.py:169: UserWarning: `grouped_entities` is deprecated and will be removed in version v5.0.0, defaulted to `aggregation_strategy=\"simple\"` instead.\n",
      "  warnings.warn(\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:94: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  final_df = final_df.append(disease_df) # adding the disease_df to existing\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:100: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  master_df = master_df.append(final_df)\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/transformers/pipelines/token_classification.py:169: UserWarning: `grouped_entities` is deprecated and will be removed in version v5.0.0, defaulted to `aggregation_strategy=\"simple\"` instead.\n",
      "  warnings.warn(\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:94: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  final_df = final_df.append(disease_df) # adding the disease_df to existing\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:100: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  master_df = master_df.append(final_df)\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/transformers/pipelines/token_classification.py:169: UserWarning: `grouped_entities` is deprecated and will be removed in version v5.0.0, defaulted to `aggregation_strategy=\"simple\"` instead.\n",
      "  warnings.warn(\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:94: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  final_df = final_df.append(disease_df) # adding the disease_df to existing\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:100: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  master_df = master_df.append(final_df)\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/transformers/pipelines/token_classification.py:169: UserWarning: `grouped_entities` is deprecated and will be removed in version v5.0.0, defaulted to `aggregation_strategy=\"simple\"` instead.\n",
      "  warnings.warn(\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:94: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  final_df = final_df.append(disease_df) # adding the disease_df to existing\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:100: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  master_df = master_df.append(final_df)\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/transformers/pipelines/token_classification.py:169: UserWarning: `grouped_entities` is deprecated and will be removed in version v5.0.0, defaulted to `aggregation_strategy=\"simple\"` instead.\n",
      "  warnings.warn(\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:94: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  final_df = final_df.append(disease_df) # adding the disease_df to existing\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:100: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  master_df = master_df.append(final_df)\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/transformers/pipelines/token_classification.py:169: UserWarning: `grouped_entities` is deprecated and will be removed in version v5.0.0, defaulted to `aggregation_strategy=\"simple\"` instead.\n",
      "  warnings.warn(\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:94: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  final_df = final_df.append(disease_df) # adding the disease_df to existing\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:100: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  master_df = master_df.append(final_df)\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/transformers/pipelines/token_classification.py:169: UserWarning: `grouped_entities` is deprecated and will be removed in version v5.0.0, defaulted to `aggregation_strategy=\"simple\"` instead.\n",
      "  warnings.warn(\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:94: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  final_df = final_df.append(disease_df) # adding the disease_df to existing\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:100: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  master_df = master_df.append(final_df)\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/transformers/pipelines/token_classification.py:169: UserWarning: `grouped_entities` is deprecated and will be removed in version v5.0.0, defaulted to `aggregation_strategy=\"simple\"` instead.\n",
      "  warnings.warn(\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:94: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  final_df = final_df.append(disease_df) # adding the disease_df to existing\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:100: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  master_df = master_df.append(final_df)\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/transformers/pipelines/token_classification.py:169: UserWarning: `grouped_entities` is deprecated and will be removed in version v5.0.0, defaulted to `aggregation_strategy=\"simple\"` instead.\n",
      "  warnings.warn(\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:94: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  final_df = final_df.append(disease_df) # adding the disease_df to existing\n",
      "/home/codespace/.python/current/lib/python3.10/site-packages/Bio_Epidemiology_NER/bio_recognizer.py:100: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  master_df = master_df.append(final_df)\n"
     ]
    }
   ],
   "source": [
    "print(diagnosis_dict)\n",
    "print(count_dict)\n",
    "for key, val in count_dict.items():\n",
    "    print(f\"{key}: {ner_prediction(str(key), compute='cpu')['entity_group'].iloc[0]}: {val}\" + '\\n')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
